<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Belinda Cornes" />

<meta name="date" content="2020-11-13" />

<title>Step 1: Sample QC [b6.v265d]</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="site_libs/d3-4.9.0/d3.min.js"></script>
<link href="site_libs/colorbrewer-1.0.0/colorbrewer.css" rel="stylesheet" />
<script src="site_libs/colorbrewer-1.0.0/colorbrewer.js"></script>
<script src="site_libs/d3-tip-0.7.1/index-min.js"></script>
<link href="site_libs/d3panels-1.4.9/d3panels.min.css" rel="stylesheet" />
<script src="site_libs/d3panels-1.4.9/d3panels.min.js"></script>
<script src="site_libs/qtlcharts_iplot-0.11.6/iplot.js"></script>
<script src="site_libs/iplot-binding-0.11.6/iplot.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">John/Tolman Workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Step 1: Sample QC [b6.v265d]</h1>
<h4 class="author">Belinda Cornes</h4>
<h4 class="date">2020-11-13</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#loading-project">Loading Project</a></li>
<li><a href="#missing-data">Missing Data</a></li>
<li><a href="#sex">Sex</a></li>
<li><a href="#sample-duplicates">Sample Duplicates</a></li>
<li><a href="#array-intensities">Array Intensities</a></li>
<li><a href="#genotype-frequencies">Genotype Frequencies</a></li>
<li><a href="#crossover-counts">Crossover Counts</a></li>
<li><a href="#genotyping-error-lod-scores">Genotyping Error LOD Scores</a></li>
<li><a href="#apparent-genotyping-errors">Apparent Genotyping Errors</a></li>
<li><a href="#removing-samples">Removing Samples</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>John-Toleman_workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryJohnTolmanWorkflowtree20c3f3ede27517da5c73300523eefe9f2a5228f5targetblank20c3f3ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/tree/20c3f3ede27517da5c73300523eefe9f2a5228f5" target="_blank">20c3f3e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryJohnTolmanWorkflowtree20c3f3ede27517da5c73300523eefe9f2a5228f5targetblank20c3f3ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/tree/20c3f3ede27517da5c73300523eefe9f2a5228f5" target="_blank">20c3f3e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  analysis/2.1_geneseek2qtl2.R
    Untracked:  analysis/2.2_geneseek2intensity.R
    Untracked:  data/bad_markers_all_129Sj.v265d.RData
    Untracked:  data/bad_markers_all_b6.v265d.RData
    Untracked:  data/bad_markers_all_d2.q105x.RData
    Untracked:  data/e_129Sj.v265d.RData
    Untracked:  data/e_b6.v265d.RData
    Untracked:  data/e_d2.q105x.RData
    Untracked:  data/e_g_snpg_samqc_129Sj.v265d.RData
    Untracked:  data/e_g_snpg_samqc_b6.v265d.RData
    Untracked:  data/e_g_snpg_samqc_d2.q105x.RData
    Untracked:  data/errors_ind.RData
    Untracked:  data/genotype_errors_marker_129Sj.v265d.RData
    Untracked:  data/genotype_errors_marker_b6.v265d.RData
    Untracked:  data/genotype_errors_marker_d2.q105x.RData
    Untracked:  data/genotype_freq_marker_129Sj.v265d.RData
    Untracked:  data/genotype_freq_marker_b6.v265d.RData
    Untracked:  data/genotype_freq_marker_d2.q105x.RData
    Untracked:  data/gm_allqc_129Sj.v265d.RData
    Untracked:  data/gm_allqc_b6.v265d.RData
    Untracked:  data/gm_allqc_d2.q105x.RData
    Untracked:  data/gm_samqc_129Sj.v265d.RData
    Untracked:  data/gm_samqc_b6.v265d.RData
    Untracked:  data/gm_samqc_d2.q105x.RData
    Untracked:  data/m_129Sj.v265d.RData
    Untracked:  data/m_b6.v265d.RData
    Untracked:  data/m_d2.q105x.RData
    Untracked:  data/nxo_129Sj.v265d.RData
    Untracked:  data/nxo_b6.v265d.RData
    Untracked:  data/nxo_d2.q105x.RData
    Untracked:  data/percent_missing_id_129Sj.v265d.RData
    Untracked:  data/percent_missing_id_b6.v265d.RData
    Untracked:  data/percent_missing_id_d2.q105x.RData
    Untracked:  data/percent_missing_marker.RData
    Untracked:  data/percent_missing_marker_129Sj.v265d.RData
    Untracked:  data/percent_missing_marker_b6.v265d.RData
    Untracked:  data/percent_missing_marker_d2.q105x.RData
    Untracked:  data/qc_info_bad_sample_129Sj.v265d.RData
    Untracked:  data/qc_info_bad_sample_b6.v265d.RData
    Untracked:  data/qc_info_bad_sample_d2.q105x.RData
    Untracked:  data/summary.cg_129Sj.v265d.RData
    Untracked:  data/summary.cg_b6.v265d.RData
    Untracked:  data/summary.cg_d2.q105x.RData
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_129Sj.v265d.pdf
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_b6.v265d.pdf
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_d2.q105x.pdf
    Untracked:  output/Percent_missing_genotype_data_129Sj.v265d.pdf
    Untracked:  output/Percent_missing_genotype_data_b6.v265d.pdf
    Untracked:  output/Percent_missing_genotype_data_d2.q105x.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_129Sj.v265d.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_b6.v265d.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_d2.q105x.pdf
    Untracked:  output/genotype_error_marker_129Sj.v265d.pdf
    Untracked:  output/genotype_error_marker_b6.v265d.pdf
    Untracked:  output/genotype_error_marker_d2.q105x.pdf
    Untracked:  output/genotype_frequency_marker.pdf
    Untracked:  output/genotype_frequency_marker_129Sj.v265d.pdf
    Untracked:  output/genotype_frequency_marker_b6.v265d.pdf
    Untracked:  output/genotype_frequency_marker_d2.q105x.pdf
    Untracked:  output/number_crossover_129Sj.v265d.pdf
    Untracked:  output/number_crossover_b6.v265d.pdf
    Untracked:  output/number_crossover_d2.q105x.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/2.1_sample_bqc_b6.v265d.Rmd</code>) and HTML (<code>docs/2.1_sample_bqc_b6.v265d.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/blob/20c3f3ede27517da5c73300523eefe9f2a5228f5/analysis/2.1_sample_bqc_b6.v265d.Rmd" target="_blank">20c3f3e</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-13
</td>
<td>
sample qc analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheJacksonLaboratory/John-Tolman_Workflow/575cd8cd86ca14e8e7917782b914120389cd82c1/docs/2.1_sample_bqc_b6.v265d.html" target="_blank">575cd8c</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/blob/4d36b4394a15d54d1fc55bec20810f21f3184378/analysis/2.1_sample_bqc_b6.v265d.Rmd" target="_blank">4d36b43</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-12
</td>
<td>
Build site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This script is running genotype QC on raw data (with some outcomes already seen in the <a href="summary.html">project at a glace</a>). Here, we first load the R/qtl2 package and the data. We’ll also load the R/broman package for some utilities and plotting functions, and R/qtlcharts for interactive graphs.</p>
<p>We will follow the steps by <code>Karl Broman</code> found <a href="https://kbroman.org/qtl2/assets/vignettes/do_diagnostics.html">here</a></p>
<div id="loading-project" class="section level3">
<h3>Loading Project</h3>
<pre class="r"><code>gm &lt;- read_cross2(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/haplo.reconst/2020-11-11/b6.v265d/all_samples_sorted/qtl2/gm_sorted.json&quot;)

#gm &lt;- get(load(&quot;gm_DO.RData&quot;))
gm</code></pre>
<pre><code>Object of class cross2 (crosstype &quot;dof1&quot;)

Total individuals               202
No. genotyped individuals       202
No. phenotyped individuals      202
No. with both geno &amp; pheno      202

No. phenotypes                    1
No. covariates                    3
No. phenotype covariates          0

No. chromosomes                  20
Total markers                137302

No. markers by chr:
    1     2     3     4     5     6     7     8     9    10    11    12    13 
10423 10441  8206  7955  8030  8130  7760  6717  6984  6631  7433  6444  6327 
   14    15    16    17    18    19     X 
 6230  5534  5179  5323  4787  3676  5092 </code></pre>
</div>
<div id="missing-data" class="section level3">
<h3>Missing Data</h3>
<pre class="r"><code>percent_missing &lt;- n_missing(gm, &quot;ind&quot;, &quot;prop&quot;)*100

labels &lt;- paste0(names(percent_missing), &quot; (&quot;, round(percent_missing,2), &quot;%)&quot;)
iplot(seq_along(percent_missing), percent_missing, indID=labels,
      chartOpts=list(xlab=&quot;Mouse&quot;, ylab=&quot;Percent missing genotype data&quot;,
                     ylim=c(0, 70)))</code></pre>
<pre><code>Set screen size to height=700 x width=1000</code></pre>
<div id="htmlwidget-23f87cbf4874345f7837" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-23f87cbf4874345f7837">{"x":{"data":{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202],"y":[3.4959,3.4806,3.2061,4.3372,4.3233,3.1005,3.0954,3.0597,3.5156,3.3539,3.1383,3.2796,3.1099,3.1697,3.1711,3.1646,3.2039,3.1413,3.1718,3.5127,3.1318,3.1682,30.037,3.0786,15.72,17.465,3.3204,5.2177,4.4049,14.187,3.0954,30.445,3.2549,3.1216,25.8,3.1281,3.4661,3.1311,3.5673,3.1529,3.0764,3.2927,3.2083,6.1987,3.225,3.2855,4.6234,3.3153,4.1361,3.1806,3.2592,3.2024,12.356,10.818,3.2075,3.2155,3.1646,3.5134,3.1667,3.5156,3.2061,3.0888,3.3736,3.1769,3.1325,3.209,3.182,3.1573,3.2155,3.1449,3.1522,3.2345,3.4289,7.289,3.7042,3.1289,3.1646,3.3364,3.5331,3.3925,3.1813,3.4384,3.528,3.1806,3.3852,3.2447,3.1879,3.3816,3.2825,3.1755,3.319,5.3568,3.1951,3.2134,3.1609,3.241,3.1158,3.3459,3.1682,3.1114,5.1063,3.1077,3.1813,3.1573,3.3241,3.2876,3.2957,3.4158,3.2847,3.3415,3.3364,6.0276,3.4034,3.1478,3.0684,3.126,4.2017,3.2556,3.2818,3.1544,3.107,18.936,3.9249,3.7785,3.1565,4.9723,3.4005,3.4974,65,3.5142,3.4683,3.5258,30.855,3.9453,20.383,3.5236,36.501,3.461,3.4843,3.3321,3.4566,3.5134,3.4442,3.4981,14.613,3.5229,3.3255,49.472,3.6402,4.5214,3.847,3.3911,3.5061,3.4617,3.4683,3.8375,3.351,6.3932,4.0276,3.3525,3.8885,3.5892,3.8849,3.6533,3.5171,3.3546,3.3415,3.5287,3.4464,3.5287,4.45,3.8091,3.3852,3.434,3.3932,3.3678,4.0364,29.714,3.9439,3.3568,3.2469,3.1995,3.2556,3.1973,3.1449,3.1143,3.3233,3.1616,3.1915,3.3983,3.1777,3.2323,3.2833,3.2658,3.1988,3.1697,3.1289,3.2825,3.4333,4.6285,4.3007,3.1507],"group":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"indID":["Jackson_Lab_John_MURGIGV01_20190728_451442_A3 (3.5%)","Jackson_Lab_John_MURGIGV01_20190728_459954_A7 (3.48%)","Jackson_Lab_John_MURGIGV01_20190728_449298_A10 (3.21%)","Jackson_Lab_John_MURGIGV01_20191031_450600_G6 (4.34%)","Jackson_Lab_John_MURGIGV01_20191031_450738_B12 (4.32%)","Jackson_Lab_John_MURGIGV01_20191031_450740_G9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_445490_G4 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_445491_F2 (3.06%)","Jackson_Lab_John_MURGIGV01_20191031_448191_D12 (3.52%)","Jackson_Lab_John_MURGIGV01_20191031_448193_D11 (3.35%)","Jackson_Lab_John_MURGIGV01_20191031_448182_B5 (3.14%)","Jackson_Lab_John_MURGIGV01_20191031_448918_C5 (3.28%)","Jackson_Lab_John_MURGIGV01_20191031_448919_A8 (3.11%)","Jackson_Lab_John_MURGIGV01_20191031_448903_A5 (3.17%)","Jackson_Lab_John_MURGIGV01_20191031_448905_A4 (3.17%)","Jackson_Lab_John_MURGIGV01_20191031_449187_E4 (3.16%)","Jackson_Lab_John_MURGIGV01_20191031_449189_G10 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_449201_A10 (3.14%)","Jackson_Lab_John_MURGIGV01_20191031_449203_F2 (3.17%)","Jackson_Lab_John_MURGIGV01_20191031_449207_E2 (3.51%)","Jackson_Lab_John_MURGIGV01_20191031_449211_H4 (3.13%)","Jackson_Lab_John_MURGIGV01_20191031_449248_A1 (3.17%)","Jackson_Lab_John_MURGIGV01_20191031_449250_B4 (30.04%)","Jackson_Lab_John_MURGIGV01_20191031_449258_A9 (3.08%)","Jackson_Lab_John_MURGIGV01_20191031_449260_G7 (15.72%)","Jackson_Lab_John_MURGIGV01_20191031_449271_G4 (17.47%)","Jackson_Lab_John_MURGIGV01_20191031_449273_D10 (3.32%)","Jackson_Lab_John_MURGIGV01_20191031_449284_B9 (5.22%)","Jackson_Lab_John_MURGIGV01_20191031_449299_D7 (4.4%)","Jackson_Lab_John_MURGIGV01_20191031_450416_D10 (14.19%)","Jackson_Lab_John_MURGIGV01_20191031_449283_F5 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_450428_E12 (30.45%)","Jackson_Lab_John_MURGIGV01_20191031_450433_G1 (3.25%)","Jackson_Lab_John_MURGIGV01_20191031_450434_D2 (3.12%)","Jackson_Lab_John_MURGIGV01_20191031_450412_A8 (25.8%)","Jackson_Lab_John_MURGIGV01_20191031_451041_F10 (3.13%)","Jackson_Lab_John_MURGIGV01_20191031_451043_A6 (3.47%)","Jackson_Lab_John_MURGIGV01_20191031_451049_C9 (3.13%)","Jackson_Lab_John_MURGIGV01_20191031_451050_C12 (3.57%)","Jackson_Lab_John_MURGIGV01_20191031_451052_C5 (3.15%)","Jackson_Lab_John_MURGIGV01_20191031_451053_F9 (3.08%)","Jackson_Lab_John_MURGIGV01_20191031_451066_H10 (3.29%)","Jackson_Lab_John_MURGIGV01_20191031_451067_E1 (3.21%)","Jackson_Lab_John_MURGIGV01_20191031_451071_H12 (6.2%)","Jackson_Lab_John_MURGIGV01_20191031_451153_H1 (3.23%)","Jackson_Lab_John_MURGIGV01_20191031_451155_E6 (3.29%)","Jackson_Lab_John_MURGIGV01_20191031_451161_F12 (4.62%)","Jackson_Lab_John_MURGIGV01_20191031_451183_F10 (3.32%)","Jackson_Lab_John_MURGIGV01_20191031_451186_D8 (4.14%)","Jackson_Lab_John_MURGIGV01_20191031_451200_D7 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_451163_B1 (3.26%)","Jackson_Lab_John_MURGIGV01_20191031_451165_G11 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451140_E11 (12.36%)","Jackson_Lab_John_MURGIGV01_20191031_451290_D1 (10.82%)","Jackson_Lab_John_MURGIGV01_20191031_451295_H4 (3.21%)","Jackson_Lab_John_MURGIGV01_20191031_451298_E5 (3.22%)","Jackson_Lab_John_MURGIGV01_20191031_451319_D9 (3.16%)","Jackson_Lab_John_MURGIGV01_20191031_451336_B8 (3.51%)","Jackson_Lab_John_MURGIGV01_20191031_451337_G8 (3.17%)","Jackson_Lab_John_MURGIGV01_20191031_451302_H6 (3.52%)","Jackson_Lab_John_MURGIGV01_20191031_454536_D6 (3.21%)","Jackson_Lab_John_MURGIGV01_20191031_454537_H9 (3.09%)","Jackson_Lab_John_MURGIGV01_20191031_451435_G3 (3.37%)","Jackson_Lab_John_MURGIGV01_20191031_451441_F3 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_451444_C7 (3.13%)","Jackson_Lab_John_MURGIGV01_20191031_451445_F5 (3.21%)","Jackson_Lab_John_MURGIGV01_20191031_451460_C10 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_451461_E7 (3.16%)","Jackson_Lab_John_MURGIGV01_20191031_451464_C7 (3.22%)","Jackson_Lab_John_MURGIGV01_20191031_451466_C4 (3.14%)","Jackson_Lab_John_MURGIGV01_20191031_451473_F11 (3.15%)","Jackson_Lab_John_MURGIGV01_20191031_451475_D3 (3.23%)","Jackson_Lab_John_MURGIGV01_20191031_451508_B6 (3.43%)","Jackson_Lab_John_MURGIGV01_20191031_451518_D7 (7.29%)","Jackson_Lab_John_MURGIGV01_20191031_451519_C12 (3.7%)","Jackson_Lab_John_MURGIGV01_20191031_451560_D4 (3.13%)","Jackson_Lab_John_MURGIGV01_20191031_451543_D1 (3.16%)","Jackson_Lab_John_MURGIGV01_20191031_451551_E12 (3.34%)","Jackson_Lab_John_MURGIGV01_20191031_451552_H6 (3.53%)","Jackson_Lab_John_MURGIGV01_20191031_451556_H7 (3.39%)","Jackson_Lab_John_MURGIGV01_20191031_451557_B9 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_451566_E7 (3.44%)","Jackson_Lab_John_MURGIGV01_20191031_451568_H11 (3.53%)","Jackson_Lab_John_MURGIGV01_20191031_455738_G5 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_451577_H2 (3.39%)","Jackson_Lab_John_MURGIGV01_20191031_451579_C5 (3.24%)","Jackson_Lab_John_MURGIGV01_20191031_451632_B3 (3.19%)","Jackson_Lab_John_MURGIGV01_20191031_451649_A11 (3.38%)","Jackson_Lab_John_MURGIGV01_20191031_451655_E1 (3.28%)","Jackson_Lab_John_MURGIGV01_20191031_451656_D4 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_455874_D8 (3.32%)","Jackson_Lab_John_MURGIGV01_20191031_455881_G8 (5.36%)","Jackson_Lab_John_MURGIGV01_20191031_455864_F2 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459907_E6 (3.21%)","Jackson_Lab_John_MURGIGV01_20191031_459909_B8 (3.16%)","Jackson_Lab_John_MURGIGV01_20191031_455899_G11 (3.24%)","Jackson_Lab_John_MURGIGV01_20191031_459901_A4 (3.12%)","Jackson_Lab_John_MURGIGV01_20191031_459915_H2 (3.35%)","Jackson_Lab_John_MURGIGV01_20191031_459922_B4 (3.17%)","Jackson_Lab_John_MURGIGV01_20191031_459923_E9 (3.11%)","Jackson_Lab_John_MURGIGV01_20191031_459934_A10 (5.11%)","Jackson_Lab_John_MURGIGV01_20191031_459936_G9 (3.11%)","Jackson_Lab_John_MURGIGV01_20191031_459943_C11 (3.18%)","Jackson_Lab_John_MURGIGV01_20191031_459969_A4 (3.16%)","Jackson_Lab_John_MURGIGV01_20191031_459959_G2 (3.32%)","Jackson_Lab_John_MURGIGV01_20191031_461048_B11 (3.29%)","Jackson_Lab_John_MURGIGV01_20191031_461036_A7 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461042_G5 (3.42%)","Jackson_Lab_John_MURGIGV01_20191031_461056_C6 (3.28%)","Jackson_Lab_John_MURGIGV01_20191031_461057_A9 (3.34%)","Jackson_Lab_John_MURGIGV01_20191031_461808_A6 (3.34%)","Jackson_Lab_John_MURGIGV01_20191031_461809_A12 (6.03%)","Jackson_Lab_John_MURGIGV01_20191031_461088_G1 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_461096_C3 (3.15%)","Jackson_Lab_John_MURGIGV01_20191031_461098_C2 (3.07%)","Jackson_Lab_John_MURGIGV01_20191031_461812_A5 (3.13%)","Jackson_Lab_John_MURGIGV01_20191031_461818_C6 (4.2%)","Jackson_Lab_John_MURGIGV01_20191031_461820_A7 (3.26%)","Jackson_Lab_John_MURGIGV01_20191031_461821_C9 (3.28%)","Jackson_Lab_John_MURGIGV01_20191031_461881_E8 (3.15%)","Jackson_Lab_John_MURGIGV01_20191031_461884_B9 (3.11%)","Jackson_Lab_John_MURGIGV01_20191031_461865_F7 (18.94%)","Jackson_Lab_John_MURGIGV01_20191031_461866_F6 (3.92%)","Jackson_Lab_John_MURGIGV01_20191031_461868_E8 (3.78%)","Jackson_Lab_John_MURGIGV01_20191031_461888_H10 (3.16%)","Jackson_Lab_John_MURGIGV01_20191107_450598_A1 (4.97%)","Jackson_Lab_John_MURGIGV01_20191107_448181_B3 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_448906_D7 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_449183_C10 (65%)","Jackson_Lab_John_MURGIGV01_20191107_449191_A10 (3.51%)","Jackson_Lab_John_MURGIGV01_20191107_449192_C9 (3.47%)","Jackson_Lab_John_MURGIGV01_20191107_449259_E4 (3.53%)","Jackson_Lab_John_MURGIGV01_20191107_449290_G1 (30.85%)","Jackson_Lab_John_MURGIGV01_20191107_450413_F10 (3.95%)","Jackson_Lab_John_MURGIGV01_20191107_449281_C12 (20.38%)","Jackson_Lab_John_MURGIGV01_20191107_450426_F5 (3.52%)","Jackson_Lab_John_MURGIGV01_20191107_450411_H12 (36.5%)","Jackson_Lab_John_MURGIGV01_20191107_451070_B7 (3.46%)","Jackson_Lab_John_MURGIGV01_20191107_451170_F4 (3.48%)","Jackson_Lab_John_MURGIGV01_20191107_451203_F11 (3.33%)","Jackson_Lab_John_MURGIGV01_20191107_451289_C8 (3.46%)","Jackson_Lab_John_MURGIGV01_20191107_451315_C7 (3.51%)","Jackson_Lab_John_MURGIGV01_20191107_451324_G3 (3.44%)","Jackson_Lab_John_MURGIGV01_20191107_451325_E7 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451305_D11 (14.61%)","Jackson_Lab_John_MURGIGV01_20191107_451500_B11 (3.52%)","Jackson_Lab_John_MURGIGV01_20191107_451510_C2 (3.33%)","Jackson_Lab_John_MURGIGV01_20191107_455737_D10 (49.47%)","Jackson_Lab_John_MURGIGV01_20191107_451626_F7 (3.64%)","Jackson_Lab_John_MURGIGV01_20191107_451627_H4 (4.52%)","Jackson_Lab_John_MURGIGV01_20191107_455877_D2 (3.85%)","Jackson_Lab_John_MURGIGV01_20191107_455882_A5 (3.39%)","Jackson_Lab_John_MURGIGV01_20191107_455865_F2 (3.51%)","Jackson_Lab_John_MURGIGV01_20191107_455898_G9 (3.46%)","Jackson_Lab_John_MURGIGV01_20191107_459904_D5 (3.47%)","Jackson_Lab_John_MURGIGV01_20191107_459913_B6 (3.84%)","Jackson_Lab_John_MURGIGV01_20191107_459933_B1 (3.35%)","Jackson_Lab_John_MURGIGV01_20191107_459951_E3 (6.39%)","Jackson_Lab_John_MURGIGV01_20191107_459942_C4 (4.03%)","Jackson_Lab_John_MURGIGV01_20191107_459970_B2 (3.35%)","Jackson_Lab_John_MURGIGV01_20191107_459961_A12 (3.89%)","Jackson_Lab_John_MURGIGV01_20191107_461047_B12 (3.59%)","Jackson_Lab_John_MURGIGV01_20191107_461035_A2 (3.88%)","Jackson_Lab_John_MURGIGV01_20191107_461043_C7 (3.65%)","Jackson_Lab_John_MURGIGV01_20191107_461829_D11 (3.52%)","Jackson_Lab_John_MURGIGV01_20191107_461832_D3 (3.35%)","Jackson_Lab_John_MURGIGV01_20191107_461089_A11 (3.34%)","Jackson_Lab_John_MURGIGV01_20191107_461814_C12 (3.53%)","Jackson_Lab_John_MURGIGV01_20191107_461819_H9 (3.45%)","Jackson_Lab_John_MURGIGV01_20191107_461878_G2 (3.53%)","Jackson_Lab_John_MURGIGV01_20191107_461880_F12 (4.45%)","Jackson_Lab_John_MURGIGV01_20191107_461867_D3 (3.81%)","Jackson_Lab_John_MURGIGV01_20191107_461889_E5 (3.39%)","Jackson_Lab_John_MURGIGV01_20191107_461906_A8 (3.43%)","Jackson_Lab_John_MURGIGV01_20191107_450631_C9 (3.39%)","Jackson_Lab_John_MURGIGV01_20191107_450652_C5 (3.37%)","Jackson_Lab_John_MURGIGV01_20191114_449206_D7 (4.04%)","Jackson_Lab_John_MURGIGV01_20191114_449262_E7 (29.71%)","Jackson_Lab_John_MURGIGV01_20191114_451160_C11 (3.94%)","Jackson_Lab_John_MURGIGV01_20191114_459939_H4 (3.36%)","Columbia_University_John_MURGIGV01_20200529_449250_G1 (3.25%)","Columbia_University_John_MURGIGV01_20200529_449260_D11 (3.2%)","Columbia_University_John_MURGIGV01_20200529_449262_D1 (3.26%)","Columbia_University_John_MURGIGV01_20200529_449271_H1 (3.2%)","Columbia_University_John_MURGIGV01_20200529_449290_G11 (3.14%)","Columbia_University_John_MURGIGV01_20200529_450416_B8 (3.11%)","Columbia_University_John_MURGIGV01_20200529_450428_B5 (3.32%)","Columbia_University_John_MURGIGV01_20200529_450411_D3 (3.16%)","Columbia_University_John_MURGIGV01_20200529_450412_B4 (3.19%)","Columbia_University_John_MURGIGV01_20200529_451140_B12 (3.4%)","Columbia_University_John_MURGIGV01_20200529_455737_D5 (3.18%)","Columbia_University_John_MURGIGV01_20200529_451704_F9 (3.23%)","Columbia_University_John_MURGIGV01_20200529_451705_C12 (3.28%)","Columbia_University_John_MURGIGV01_20200529_451707_G11 (3.27%)","Columbia_University_John_MURGIGV01_20200529_451708_F10 (3.2%)","Columbia_University_John_MURGIGV01_20200529_461865_A5 (3.17%)","Columbia_University_John_MURGIGV01_20200529_451825_F5 (3.13%)","Columbia_University_John_MURGIGV01_20200529_449183_C8 (3.28%)","Columbia_University_John_MURGIGV01_20200529_449281_H8 (3.43%)","Columbia_University_John_MURGIGV01_20200529_451290_F12 (4.63%)","Columbia_University_John_MURGIGV01_20200529_451305_A1 (4.3%)","Columbia_University_John_MURGIGV01_20200529_451706_D4 (3.15%)"]},"chartOpts":{"xlab":"Mouse","ylab":"Percent missing genotype data","ylim":[0,70]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>#save into pdf
pdf(file = &quot;output/Percent_missing_genotype_data_b6.v265d.pdf&quot;, width = 20, height = 20)
labels &lt;- as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;V01_&quot;))[,2])
labels[percent_missing &lt; 10] = &quot;&quot;
# Change point shapes and colors
p &lt;- ggplot(data = data.frame(Mouse=seq_along(percent_missing),  
                         Percent_missing_genotype_data = percent_missing,
                         batch = factor(as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,5]))
                         ), 
        aes(x=Mouse, y=Percent_missing_genotype_data, color = batch)) +
        geom_point() +
        geom_hline(yintercept=10, linetype=&quot;solid&quot;, color = &quot;red&quot;) +
        geom_text_repel(aes(label=labels), vjust = 0, nudge_y = 0.01, show.legend = FALSE, size=3) +
        theme(text = element_text(size = 10))
p

dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="figure/2.1_sample_bqc_b6.v265d.Rmd/Missing%20data%20per%20sample-2.png" width="912" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Missing-data-per-sample-2">
Past versions of “Missing data per sample-2.png”
</button>
</p>
<div id="fig-no-spaces-Missing-data-per-sample-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/blob/575cd8cd86ca14e8e7917782b914120389cd82c1/docs/figure/2.1_sample_bqc_b6.v265d.Rmd/Missing data per sample-2.png" target="_blank">575cd8c</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>save(percent_missing,file = &quot;data/percent_missing_id_b6.v265d.RData&quot;)

gm.covar = data.frame(id=rownames(gm$covar),gm$covar)
qc_info_cr &lt;- merge(gm.covar,
                  data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = &quot;id&quot;)
bad.sample.cr &lt;- qc_info_cr[qc_info_cr$percent_missing &gt;= 10,]</code></pre>
<p>There are 16 mice that are missing &gt;10% of their genotypes. These samples are:</p>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
id
</th>
<th style="text-align:center;">
percent_missing
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
449250
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449250_B4
</td>
<td style="text-align:center;">
30.0367074041165
</td>
</tr>
<tr>
<td style="text-align:center;">
449260
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449260_G7
</td>
<td style="text-align:center;">
15.7200914771817
</td>
</tr>
<tr>
<td style="text-align:center;">
449271
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449271_G4
</td>
<td style="text-align:center;">
17.4651498157347
</td>
</tr>
<tr>
<td style="text-align:center;">
450412
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450412_A8
</td>
<td style="text-align:center;">
25.8000611790069
</td>
</tr>
<tr>
<td style="text-align:center;">
450416
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450416_D10
</td>
<td style="text-align:center;">
14.1869746981107
</td>
</tr>
<tr>
<td style="text-align:center;">
450428
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450428_E12
</td>
<td style="text-align:center;">
30.4452957713653
</td>
</tr>
<tr>
<td style="text-align:center;">
451140
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451140_E11
</td>
<td style="text-align:center;">
12.3559744213486
</td>
</tr>
<tr>
<td style="text-align:center;">
451290
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451290_D1
</td>
<td style="text-align:center;">
10.8184877132161
</td>
</tr>
<tr>
<td style="text-align:center;">
461865
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_461865_F7
</td>
<td style="text-align:center;">
18.9363592664346
</td>
</tr>
<tr>
<td style="text-align:center;">
449183
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449183_C10
</td>
<td style="text-align:center;">
64.9997815035469
</td>
</tr>
<tr>
<td style="text-align:center;">
449281
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:center;">
20.3828057857861
</td>
</tr>
<tr>
<td style="text-align:center;">
449290
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449290_G1
</td>
<td style="text-align:center;">
30.8546124601244
</td>
</tr>
<tr>
<td style="text-align:center;">
450411
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_450411_H12
</td>
<td style="text-align:center;">
36.5012891290731
</td>
</tr>
<tr>
<td style="text-align:center;">
451305
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_451305_D11
</td>
<td style="text-align:center;">
14.6130427816055
</td>
</tr>
<tr>
<td style="text-align:center;">
455737
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_455737_D10
</td>
<td style="text-align:center;">
49.4719669050706
</td>
</tr>
<tr>
<td style="text-align:center;">
449262
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191114_449262_E7
</td>
<td style="text-align:center;">
29.7140609750768
</td>
</tr>
</tbody>
</table>
</div>
<div id="sex" class="section level3">
<h3>Sex</h3>
<pre class="r"><code>hdf5_filename &lt;- dir(path = filepaths, pattern = &quot;^hdf5_*&quot;, full.names = TRUE)
snps_file &lt;- &quot;/Users/corneb/Documents/MyJax/CS/Projects/QTL2_support.files/gm_uwisc_v1.csv&quot;
snps &lt;- read.csv(snps_file)

snps &lt;- snps[snps$unique == TRUE, ]
#snps &lt;- snps[snps$chr %in% c(1:19, &quot;X&quot;), ]
snps$chr &lt;- sub(&quot;^chr&quot;, &quot;&quot;, snps$chr)  ###remove prefix &quot;chr&quot;
colnames(snps)[colnames(snps)==&quot;bp_mm10&quot;] &lt;- &quot;pos&quot; 
colnames(snps)[colnames(snps)==&quot;cM_cox&quot;] &lt;- &quot;cM&quot;
snps &lt;- snps %&gt;% drop_na(chr, marker) 
snps$pos &lt;- snps$pos * 1e-6
rownames(snps) &lt;- snps$marker
colnames(snps)[1:4] &lt;- c(&quot;marker&quot;, &quot;chr&quot;, &quot;pos&quot;, &quot;pos&quot;) 


#  g &lt;- h5read(hdf5_filename, &quot;G&quot;)
#  g &lt;- do.call(cbind, g)
x &lt;- h5read(hdf5_filename, &quot;X&quot;) # X channel intensities
x &lt;- do.call(cbind, x)
y &lt;- h5read(hdf5_filename, &quot;Y&quot;) # Y channel intensities
y &lt;- do.call(cbind, y)
rn &lt;- h5read(hdf5_filename, &quot;rownames&quot;)[[1]]  # markers 
cn &lt;- h5read(hdf5_filename, &quot;colnames&quot;)  # samples
cn &lt;- do.call(c, cn)
# dimnames(g) &lt;- list(rn, cn)
dimnames(x) &lt;- list(rn, cn)
dimnames(y) &lt;- list(rn, cn)
#cr &lt;- colMeans(g != &quot;--&quot;) # Call rate for each sample avg 0.95
#  sex &lt;- determine_sex(x = x, y = y, markers = snps)$se

markers &lt;- snps

chrx &lt;- markers$marker[which(markers$chr == &quot;X&quot;)]
chry &lt;- markers$marker[which(markers$chr == &quot;Y&quot;)]
#x[chrx,ind_ids(gm)]

chrx_int &lt;- colMeans(x[chrx,ind_ids(gm)] + y[chrx,ind_ids(gm)], na.rm = T)
chry_int &lt;- colMeans(x[chry,ind_ids(gm)] + y[chry,ind_ids(gm)], na.rm = T)
#  df &lt;- data.frame(x = chrx_int, y = chry_int)
#  mc &lt;- Mclust(data = df, G = 2, model = &quot;VVV&quot;, prior=priorControl()) 
#  # Determine which cluster is females by selecting the cluster with the
#  # higher Chr X intensity mean.
#  cluster_means &lt;- mc$parameters$mean
#  female_cl &lt;- which.max(cluster_means[&quot;x&quot;,])
#  sex &lt;- rep(&quot;M&quot;, length(mc$classification))
#  names(sex) &lt;- colnames(x)
#  sex[mc$classification == female_cl] &lt;- &quot;F&quot;
#  return(list(sex = sex))

#xint &lt;- read_csv_numer(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/Data/GM/Simon.John.Tolman_qtl2_chrXint.csv&quot;, transpose=TRUE)
#yint &lt;- read_csv_numer(&quot;data/Univ_of_Penn_Thaiss_qtl2_chrYint.csv&quot;, transpose=TRUE)

#xint2 &lt;- t(x[chrx,ind_ids(gm)])
#yint2 &lt;- t(y[chry,ind_ids(gm)])

sample_file &lt;- dir(path = filepaths, pattern = &quot;^DODB_*&quot;, full.names = TRUE)
samples &lt;- read.csv(sample_file)

all.equal(as.character(ind_ids(gm)), as.character(samples$Unique.Sample.ID))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>#sex order
sex &lt;- samples$Sex
#sex &lt;- gm$covar[rownames(xint),&quot;sexp&quot;]
#sex &lt;- substr(rownames(xint), 1, 1)

#x_pval &lt;- apply(xint2, 2, function(a) t.test(a ~ sex)$p.value)
#y_pval &lt;- apply(yint2, 2, function(a) t.test(a ~ sex)$p.value)

#xint_ave &lt;- rowMeans(xint2[, x_pval &lt; 0.05/length(x_pval)], na.rm=TRUE)
#yint_ave &lt;- rowMeans(yint2[, y_pval &lt; 0.05/length(y_pval)], na.rm=TRUE)

#BAF = yint / (xint + yint)
#LRR = log2((xint + yint) / (xint_ave+yint_ave))

point_colors &lt;- as.character( brocolors(&quot;web&quot;)[c(&quot;green&quot;, &quot;purple&quot;,&quot;red&quot;)] )
percent_missing &lt;- n_missing(gm, summary=&quot;proportion&quot;)*100
labels &lt;- paste0(names(chrx_int), &quot; (&quot;, round(percent_missing), &quot;%)&quot;)
iplot( chrx_int,  chry_int, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab=&quot;Average X chr intensity&quot;, ylab=&quot;Average Y chr intensity&quot;))</code></pre>
<div id="htmlwidget-f299aebb9a55300a4535" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-f299aebb9a55300a4535">{"x":{"data":{"x":[0.84268,0.84072,0.84758,0.83978,1.0931,1.1169,0.86473,0.86595,1.0845,1.0999,1.0775,0.83858,0.85971,0.86275,0.86965,0.85891,0.87571,0.86185,0.86521,1.134,1.1113,0.86286,1.2715,0.87294,1.102,0.97924,1.1329,1.165,0.90565,1.3554,0.85723,1.0245,1.1305,1.1072,0.99097,1.0986,1.104,1.0922,1.0894,1.0677,1.091,0.87186,0.86814,1.1177,0.85024,0.8838,0.80989,0.87541,0.88176,0.87134,1.1163,1.109,0.88915,0.84357,1.1049,1.116,0.87412,0.85955,0.84919,1.0838,0.87847,0.86851,1.1327,0.87261,1.1176,1.0998,0.86672,0.87366,1.1092,1.1242,0.87651,0.85849,0.8665,1.0713,1.0811,0.86624,1.0801,0.81546,0.83655,0.86823,0.85439,1.1158,1.123,0.87095,1.1419,1.1425,0.87154,0.867,0.87833,0.87731,1.1042,1.1265,1.1181,1.1165,1.1201,0.87108,0.86767,1.1215,1.1152,1.1207,0.90944,1.114,1.1424,0.86796,0.87181,1.1079,0.86698,0.87385,1.1176,1.1104,0.85567,0.80885,1.1098,1.1173,1.1018,0.87534,0.84927,1.1261,1.1435,1.1028,0.88225,0.99037,0.84009,1.1268,1.0968,0.85394,1.0439,1.0752,0.57583,1.1329,1.1197,0.87599,1.3407,1.1541,1.1621,1.1184,1.096,1.1017,1.0989,0.86903,0.86879,0.85362,1.1064,1.1071,1.5224,1.1009,0.84738,0.77481,0.87319,0.87988,0.85173,1.0969,1.1181,0.87109,0.8608,1.0861,0.84322,1.0259,1.1331,0.85941,0.8505,1.0963,0.85303,0.86435,1.0852,1.0915,1.1093,0.85112,0.85862,0.88284,0.88167,1.1014,1.1224,1.0873,0.86017,0.87227,1.0753,1.2694,0.82554,1.0917,0.86054,1.0595,1.0594,1.0692,1.0819,1.0853,1.1336,0.85729,0.84124,0.83663,0.85646,1.0683,1.0815,0.84595,0.84998,0.86424,0.8668,0.85762,0.86195,0.79039,1.0634,1.0772],"y":[0.69019,0.70477,0.69277,0.64416,0.093065,0.050645,0.72129,0.68984,0.086,0.10952,0.072258,0.67,0.71803,0.7129,0.69506,0.68781,0.73039,0.72239,0.73413,0.094935,0.062,0.72545,0.5619,0.73655,0.32468,0.44339,0.058065,0.15726,0.69339,0.10958,0.71248,0.13177,0.056419,0.086613,0.55884,0.070516,0.098839,0.070548,0.055968,0.077484,0.054516,0.72829,0.72268,0.12539,0.70397,0.74974,0.6189,0.69629,0.72171,0.74081,0.072645,0.07171,0.69113,0.65868,0.047581,0.055548,0.73074,0.72329,0.70771,0.081065,0.72074,0.72881,0.069677,0.7061,0.048484,0.047613,0.72,0.71226,0.078484,0.046387,0.74281,0.72045,0.69397,0.062645,0.073581,0.71681,0.076484,0.65381,0.66797,0.72632,0.70935,0.065581,0.073484,0.71152,0.080387,0.058065,0.7151,0.71416,0.73877,0.73035,0.074903,0.13671,0.057226,0.074548,0.070613,0.71787,0.73252,0.051871,0.059839,0.053065,0.72465,0.066355,0.079774,0.74365,0.72152,0.07929,0.71577,0.7101,0.053871,0.055194,0.69684,0.53484,0.089258,0.045742,0.04929,0.72248,0.64668,0.072032,0.075355,0.067968,0.74058,0.56171,0.68516,0.059032,0.053129,0.67165,0.080903,0.073742,0.74906,0.089419,0.096516,0.69745,0.19945,0.099065,1.1314,0.10274,0.48206,0.089645,0.10316,0.70403,0.72558,0.67006,0.089484,0.092742,1.1833,0.095355,0.67713,0.71606,0.69142,0.67184,0.6761,0.087677,0.10313,0.69103,0.7089,0.094065,0.70313,0.106,0.11948,0.71271,0.6651,0.098968,0.67577,0.70126,0.10526,0.087258,0.098355,0.64545,0.68606,0.70503,0.66855,0.11074,0.09571,0.078742,0.70326,0.7121,0.073903,0.17432,0.61026,0.04429,0.72842,0.057548,0.036548,0.11084,0.090032,0.041613,0.090419,0.71474,0.68406,0.67435,0.71987,0.049742,0.067065,0.69465,0.7031,0.72394,0.73084,0.7079,0.70177,0.58652,0.061032,0.054903],"group":[2,2,2,2,1,1,2,2,1,1,1,2,2,2,2,2,2,2,2,1,1,2,2,2,1,1,1,1,2,1,2,1,1,1,2,1,1,1,1,1,1,2,2,1,2,2,2,2,2,2,1,1,2,2,1,1,2,2,2,1,2,2,1,2,1,1,2,2,1,1,2,2,2,1,1,2,1,2,2,2,2,1,1,2,1,1,2,2,2,2,1,1,1,1,1,2,2,1,1,1,2,1,1,2,2,1,2,2,1,1,2,2,1,1,1,2,2,1,1,1,2,2,2,1,1,2,1,1,2,1,1,2,1,1,2,1,2,1,1,2,2,2,1,1,1,1,2,2,2,2,2,1,1,2,2,1,2,1,1,2,2,1,2,2,1,1,1,2,2,2,2,1,1,1,2,2,1,1,2,1,2,1,1,1,1,1,1,2,2,2,2,1,1,2,2,2,2,2,2,2,1,1],"indID":["Jackson_Lab_John_MURGIGV01_20190728_451442_A3 (3%)","Jackson_Lab_John_MURGIGV01_20190728_459954_A7 (3%)","Jackson_Lab_John_MURGIGV01_20190728_449298_A10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450600_G6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_450738_B12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_450740_G9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_445490_G4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_445491_F2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448191_D12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_448193_D11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448182_B5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448918_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448919_A8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448903_A5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448905_A4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449187_E4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449189_G10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449201_A10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449203_F2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449207_E2 (4%)","Jackson_Lab_John_MURGIGV01_20191031_449211_H4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449248_A1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449250_B4 (30%)","Jackson_Lab_John_MURGIGV01_20191031_449258_A9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449260_G7 (16%)","Jackson_Lab_John_MURGIGV01_20191031_449271_G4 (17%)","Jackson_Lab_John_MURGIGV01_20191031_449273_D10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449284_B9 (5%)","Jackson_Lab_John_MURGIGV01_20191031_449299_D7 (4%)","Jackson_Lab_John_MURGIGV01_20191031_450416_D10 (14%)","Jackson_Lab_John_MURGIGV01_20191031_449283_F5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450428_E12 (30%)","Jackson_Lab_John_MURGIGV01_20191031_450433_G1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450434_D2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450412_A8 (26%)","Jackson_Lab_John_MURGIGV01_20191031_451041_F10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451043_A6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451049_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451050_C12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451052_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451053_F9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451066_H10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451067_E1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451071_H12 (6%)","Jackson_Lab_John_MURGIGV01_20191031_451153_H1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451155_E6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451161_F12 (5%)","Jackson_Lab_John_MURGIGV01_20191031_451183_F10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451186_D8 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451200_D7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451163_B1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451165_G11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451140_E11 (12%)","Jackson_Lab_John_MURGIGV01_20191031_451290_D1 (11%)","Jackson_Lab_John_MURGIGV01_20191031_451295_H4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451298_E5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451319_D9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451336_B8 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451337_G8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451302_H6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_454536_D6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_454537_H9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451435_G3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451441_F3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451444_C7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451445_F5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451460_C10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451461_E7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451464_C7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451466_C4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451473_F11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451475_D3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451508_B6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451518_D7 (7%)","Jackson_Lab_John_MURGIGV01_20191031_451519_C12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451560_D4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451543_D1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451551_E12 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451552_H6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451556_H7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451557_B9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451566_E7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451568_H11 (4%)","Jackson_Lab_John_MURGIGV01_20191031_455738_G5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451577_H2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451579_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451632_B3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451649_A11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451655_E1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451656_D4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_455874_D8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_455881_G8 (5%)","Jackson_Lab_John_MURGIGV01_20191031_455864_F2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459907_E6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459909_B8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_455899_G11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459901_A4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459915_H2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459922_B4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459923_E9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459934_A10 (5%)","Jackson_Lab_John_MURGIGV01_20191031_459936_G9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459943_C11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459969_A4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459959_G2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461048_B11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461036_A7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461042_G5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461056_C6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461057_A9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461808_A6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461809_A12 (6%)","Jackson_Lab_John_MURGIGV01_20191031_461088_G1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461096_C3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461098_C2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461812_A5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461818_C6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_461820_A7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461821_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461881_E8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461884_B9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461865_F7 (19%)","Jackson_Lab_John_MURGIGV01_20191031_461866_F6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_461868_E8 (4%)","Jackson_Lab_John_MURGIGV01_20191031_461888_H10 (3%)","Jackson_Lab_John_MURGIGV01_20191107_450598_A1 (5%)","Jackson_Lab_John_MURGIGV01_20191107_448181_B3 (3%)","Jackson_Lab_John_MURGIGV01_20191107_448906_D7 (3%)","Jackson_Lab_John_MURGIGV01_20191107_449183_C10 (65%)","Jackson_Lab_John_MURGIGV01_20191107_449191_A10 (4%)","Jackson_Lab_John_MURGIGV01_20191107_449192_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_449259_E4 (4%)","Jackson_Lab_John_MURGIGV01_20191107_449290_G1 (31%)","Jackson_Lab_John_MURGIGV01_20191107_450413_F10 (4%)","Jackson_Lab_John_MURGIGV01_20191107_449281_C12 (20%)","Jackson_Lab_John_MURGIGV01_20191107_450426_F5 (4%)","Jackson_Lab_John_MURGIGV01_20191107_450411_H12 (37%)","Jackson_Lab_John_MURGIGV01_20191107_451070_B7 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451170_F4 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451203_F11 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451289_C8 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451315_C7 (4%)","Jackson_Lab_John_MURGIGV01_20191107_451324_G3 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451325_E7 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451305_D11 (15%)","Jackson_Lab_John_MURGIGV01_20191107_451500_B11 (4%)","Jackson_Lab_John_MURGIGV01_20191107_451510_C2 (3%)","Jackson_Lab_John_MURGIGV01_20191107_455737_D10 (49%)","Jackson_Lab_John_MURGIGV01_20191107_451626_F7 (4%)","Jackson_Lab_John_MURGIGV01_20191107_451627_H4 (5%)","Jackson_Lab_John_MURGIGV01_20191107_455877_D2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_455882_A5 (3%)","Jackson_Lab_John_MURGIGV01_20191107_455865_F2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_455898_G9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459904_D5 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459913_B6 (4%)","Jackson_Lab_John_MURGIGV01_20191107_459933_B1 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459951_E3 (6%)","Jackson_Lab_John_MURGIGV01_20191107_459942_C4 (4%)","Jackson_Lab_John_MURGIGV01_20191107_459970_B2 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459961_A12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461047_B12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461035_A2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461043_C7 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461829_D11 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461832_D3 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461089_A11 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461814_C12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461819_H9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461878_G2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461880_F12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461867_D3 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461889_E5 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461906_A8 (3%)","Jackson_Lab_John_MURGIGV01_20191107_450631_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_450652_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191114_449206_D7 (4%)","Jackson_Lab_John_MURGIGV01_20191114_449262_E7 (30%)","Jackson_Lab_John_MURGIGV01_20191114_451160_C11 (4%)","Jackson_Lab_John_MURGIGV01_20191114_459939_H4 (3%)","Columbia_University_John_MURGIGV01_20200529_449250_G1 (3%)","Columbia_University_John_MURGIGV01_20200529_449260_D11 (3%)","Columbia_University_John_MURGIGV01_20200529_449262_D1 (3%)","Columbia_University_John_MURGIGV01_20200529_449271_H1 (3%)","Columbia_University_John_MURGIGV01_20200529_449290_G11 (3%)","Columbia_University_John_MURGIGV01_20200529_450416_B8 (3%)","Columbia_University_John_MURGIGV01_20200529_450428_B5 (3%)","Columbia_University_John_MURGIGV01_20200529_450411_D3 (3%)","Columbia_University_John_MURGIGV01_20200529_450412_B4 (3%)","Columbia_University_John_MURGIGV01_20200529_451140_B12 (3%)","Columbia_University_John_MURGIGV01_20200529_455737_D5 (3%)","Columbia_University_John_MURGIGV01_20200529_451704_F9 (3%)","Columbia_University_John_MURGIGV01_20200529_451705_C12 (3%)","Columbia_University_John_MURGIGV01_20200529_451707_G11 (3%)","Columbia_University_John_MURGIGV01_20200529_451708_F10 (3%)","Columbia_University_John_MURGIGV01_20200529_461865_A5 (3%)","Columbia_University_John_MURGIGV01_20200529_451825_F5 (3%)","Columbia_University_John_MURGIGV01_20200529_449183_C8 (3%)","Columbia_University_John_MURGIGV01_20200529_449281_H8 (3%)","Columbia_University_John_MURGIGV01_20200529_451290_F12 (5%)","Columbia_University_John_MURGIGV01_20200529_451305_A1 (4%)","Columbia_University_John_MURGIGV01_20200529_451706_D4 (3%)"]},"chartOpts":{"pointcolor":["#2ecc40","#b10dc9","#ff4136"],"pointsize":4,"xlab":"Average X chr intensity","ylab":"Average Y chr intensity"}},"evals":[],"jsHooks":[]}</script>
<p>For figures above and below, those labelled as female in metadata given, are coloured <code>green</code>, with those labelled as male are coloured as <code>purple</code>. The above is an interactive scatterplot of the average SNP intensity on the Y chromosome versus the average SNP intensity on the X chromosome.</p>
<p>The cluster in the upper-left are male mice (low X chromosome intensity and high Y chromosome intensity), and the cluster in the lower-right are female mice (high X chromosome intensity and low Y chromsome intensity).</p>
<pre class="r"><code>phetX &lt;- rowSums(gm$geno$X == 2)/rowSums(gm$geno$X != 0)
phetX &lt;- phetX[names(phetX) %in% names(chrx_int)]
iplot(chrx_int, phetX, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab=&quot;Average X chr intensity&quot;, ylab=&quot;Proportion het on X chr&quot;))</code></pre>
<div id="htmlwidget-38fde9aaaa8acc82f1a4" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-38fde9aaaa8acc82f1a4">{"x":{"data":{"x":[0.84268,0.84072,0.84758,0.83978,1.0931,1.1169,0.86473,0.86595,1.0845,1.0999,1.0775,0.83858,0.85971,0.86275,0.86965,0.85891,0.87571,0.86185,0.86521,1.134,1.1113,0.86286,1.2715,0.87294,1.102,0.97924,1.1329,1.165,0.90565,1.3554,0.85723,1.0245,1.1305,1.1072,0.99097,1.0986,1.104,1.0922,1.0894,1.0677,1.091,0.87186,0.86814,1.1177,0.85024,0.8838,0.80989,0.87541,0.88176,0.87134,1.1163,1.109,0.88915,0.84357,1.1049,1.116,0.87412,0.85955,0.84919,1.0838,0.87847,0.86851,1.1327,0.87261,1.1176,1.0998,0.86672,0.87366,1.1092,1.1242,0.87651,0.85849,0.8665,1.0713,1.0811,0.86624,1.0801,0.81546,0.83655,0.86823,0.85439,1.1158,1.123,0.87095,1.1419,1.1425,0.87154,0.867,0.87833,0.87731,1.1042,1.1265,1.1181,1.1165,1.1201,0.87108,0.86767,1.1215,1.1152,1.1207,0.90944,1.114,1.1424,0.86796,0.87181,1.1079,0.86698,0.87385,1.1176,1.1104,0.85567,0.80885,1.1098,1.1173,1.1018,0.87534,0.84927,1.1261,1.1435,1.1028,0.88225,0.99037,0.84009,1.1268,1.0968,0.85394,1.0439,1.0752,0.57583,1.1329,1.1197,0.87599,1.3407,1.1541,1.1621,1.1184,1.096,1.1017,1.0989,0.86903,0.86879,0.85362,1.1064,1.1071,1.5224,1.1009,0.84738,0.77481,0.87319,0.87988,0.85173,1.0969,1.1181,0.87109,0.8608,1.0861,0.84322,1.0259,1.1331,0.85941,0.8505,1.0963,0.85303,0.86435,1.0852,1.0915,1.1093,0.85112,0.85862,0.88284,0.88167,1.1014,1.1224,1.0873,0.86017,0.87227,1.0753,1.2694,0.82554,1.0917,0.86054,1.0595,1.0594,1.0692,1.0819,1.0853,1.1336,0.85729,0.84124,0.83663,0.85646,1.0683,1.0815,0.84595,0.84998,0.86424,0.8668,0.85762,0.86195,0.79039,1.0634,1.0772],"y":[0.011633,0.014363,0.010797,0.016248,0.28783,0.27101,0.0099482,0.010151,0.14585,0.163,0.14161,0.011051,0.015641,0.015388,0.012041,0.0097551,0.0097208,0.010162,0.012622,0.17366,0.22485,0.0091135,0.3581,0.0089045,0.15248,0.22456,0.12174,0.2287,0.017936,0.22821,0.012023,0.3561,0.27259,0.19042,0.29442,0.11655,0.13371,0.14668,0.13207,0.086471,0.091304,0.011823,0.009751,0.27707,0.010601,0.010596,0.020323,0.012492,0.014499,0.009973,0.20492,0.215,0.077384,0.067542,0.13636,0.11422,0.011843,0.011012,0.011194,0.18021,0.0091135,0.012011,0.13722,0.011196,0.26739,0.20083,0.013071,0.010141,0.25083,0.24829,0.013071,0.013136,0.010266,0.1849,0.15759,0.012903,0.12373,0.01357,0.013423,0.017112,0.012256,0.11378,0.20241,0.011633,0.20237,0.18389,0.0091192,0.010391,0.010991,0.010757,0.14452,0.14953,0.21213,0.29693,0.22866,0.011638,0.011601,0.32319,0.23983,0.2691,0.027838,0.27771,0.35173,0.014087,0.012911,0.1705,0.010618,0.013073,0.16438,0.1796,0.010196,0.028327,0.24688,0.21235,0.21433,0.011192,0.020946,0.25358,0.34125,0.084093,0.0076668,0.17137,0.01315,0.095436,0.16912,0.031632,0.14141,0.092512,0.21178,0.29819,0.18702,0.012343,0.29219,0.28121,0.025956,0.20445,0.51467,0.23901,0.23523,0.011243,0.014605,0.012279,0.22151,0.25036,0.011705,0.16501,0.0099792,0.094483,0.012834,0.018335,0.013414,0.12513,0.29038,0.012523,0.015455,0.19563,0.016942,0.11073,0.19662,0.010806,0.01489,0.11231,0.016906,0.014005,0.12653,0.16265,0.24807,0.014055,0.019058,0.012938,0.016975,0.093101,0.21196,0.21734,0.011492,0.016308,0.22033,0.27796,0.017053,0.31769,0.0095436,0.13114,0.13144,0.117,0.13395,0.21805,0.20838,0.009338,0.010557,0.011411,0.011823,0.14529,0.13568,0.0099482,0.011231,0.0093226,0.014105,0.015378,0.012036,0.021538,0.29697,0.21128],"group":[2,2,2,2,1,1,2,2,1,1,1,2,2,2,2,2,2,2,2,1,1,2,2,2,1,1,1,1,2,1,2,1,1,1,2,1,1,1,1,1,1,2,2,1,2,2,2,2,2,2,1,1,2,2,1,1,2,2,2,1,2,2,1,2,1,1,2,2,1,1,2,2,2,1,1,2,1,2,2,2,2,1,1,2,1,1,2,2,2,2,1,1,1,1,1,2,2,1,1,1,2,1,1,2,2,1,2,2,1,1,2,2,1,1,1,2,2,1,1,1,2,2,2,1,1,2,1,1,2,1,1,2,1,1,2,1,2,1,1,2,2,2,1,1,1,1,2,2,2,2,2,1,1,2,2,1,2,1,1,2,2,1,2,2,1,1,1,2,2,2,2,1,1,1,2,2,1,1,2,1,2,1,1,1,1,1,1,2,2,2,2,1,1,2,2,2,2,2,2,2,1,1],"indID":["Jackson_Lab_John_MURGIGV01_20190728_451442_A3 (3%)","Jackson_Lab_John_MURGIGV01_20190728_459954_A7 (3%)","Jackson_Lab_John_MURGIGV01_20190728_449298_A10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450600_G6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_450738_B12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_450740_G9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_445490_G4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_445491_F2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448191_D12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_448193_D11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448182_B5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448918_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448919_A8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448903_A5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_448905_A4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449187_E4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449189_G10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449201_A10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449203_F2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449207_E2 (4%)","Jackson_Lab_John_MURGIGV01_20191031_449211_H4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449248_A1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449250_B4 (30%)","Jackson_Lab_John_MURGIGV01_20191031_449258_A9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449260_G7 (16%)","Jackson_Lab_John_MURGIGV01_20191031_449271_G4 (17%)","Jackson_Lab_John_MURGIGV01_20191031_449273_D10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_449284_B9 (5%)","Jackson_Lab_John_MURGIGV01_20191031_449299_D7 (4%)","Jackson_Lab_John_MURGIGV01_20191031_450416_D10 (14%)","Jackson_Lab_John_MURGIGV01_20191031_449283_F5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450428_E12 (30%)","Jackson_Lab_John_MURGIGV01_20191031_450433_G1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450434_D2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_450412_A8 (26%)","Jackson_Lab_John_MURGIGV01_20191031_451041_F10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451043_A6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451049_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451050_C12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451052_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451053_F9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451066_H10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451067_E1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451071_H12 (6%)","Jackson_Lab_John_MURGIGV01_20191031_451153_H1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451155_E6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451161_F12 (5%)","Jackson_Lab_John_MURGIGV01_20191031_451183_F10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451186_D8 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451200_D7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451163_B1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451165_G11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451140_E11 (12%)","Jackson_Lab_John_MURGIGV01_20191031_451290_D1 (11%)","Jackson_Lab_John_MURGIGV01_20191031_451295_H4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451298_E5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451319_D9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451336_B8 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451337_G8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451302_H6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_454536_D6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_454537_H9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451435_G3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451441_F3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451444_C7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451445_F5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451460_C10 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451461_E7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451464_C7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451466_C4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451473_F11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451475_D3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451508_B6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451518_D7 (7%)","Jackson_Lab_John_MURGIGV01_20191031_451519_C12 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451560_D4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451543_D1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451551_E12 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451552_H6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_451556_H7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451557_B9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451566_E7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451568_H11 (4%)","Jackson_Lab_John_MURGIGV01_20191031_455738_G5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451577_H2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451579_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451632_B3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451649_A11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451655_E1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_451656_D4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_455874_D8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_455881_G8 (5%)","Jackson_Lab_John_MURGIGV01_20191031_455864_F2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459907_E6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459909_B8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_455899_G11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459901_A4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459915_H2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459922_B4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459923_E9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459934_A10 (5%)","Jackson_Lab_John_MURGIGV01_20191031_459936_G9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459943_C11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459969_A4 (3%)","Jackson_Lab_John_MURGIGV01_20191031_459959_G2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461048_B11 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461036_A7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461042_G5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461056_C6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461057_A9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461808_A6 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461809_A12 (6%)","Jackson_Lab_John_MURGIGV01_20191031_461088_G1 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461096_C3 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461098_C2 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461812_A5 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461818_C6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_461820_A7 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461821_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461881_E8 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461884_B9 (3%)","Jackson_Lab_John_MURGIGV01_20191031_461865_F7 (19%)","Jackson_Lab_John_MURGIGV01_20191031_461866_F6 (4%)","Jackson_Lab_John_MURGIGV01_20191031_461868_E8 (4%)","Jackson_Lab_John_MURGIGV01_20191031_461888_H10 (3%)","Jackson_Lab_John_MURGIGV01_20191107_450598_A1 (5%)","Jackson_Lab_John_MURGIGV01_20191107_448181_B3 (3%)","Jackson_Lab_John_MURGIGV01_20191107_448906_D7 (3%)","Jackson_Lab_John_MURGIGV01_20191107_449183_C10 (65%)","Jackson_Lab_John_MURGIGV01_20191107_449191_A10 (4%)","Jackson_Lab_John_MURGIGV01_20191107_449192_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_449259_E4 (4%)","Jackson_Lab_John_MURGIGV01_20191107_449290_G1 (31%)","Jackson_Lab_John_MURGIGV01_20191107_450413_F10 (4%)","Jackson_Lab_John_MURGIGV01_20191107_449281_C12 (20%)","Jackson_Lab_John_MURGIGV01_20191107_450426_F5 (4%)","Jackson_Lab_John_MURGIGV01_20191107_450411_H12 (37%)","Jackson_Lab_John_MURGIGV01_20191107_451070_B7 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451170_F4 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451203_F11 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451289_C8 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451315_C7 (4%)","Jackson_Lab_John_MURGIGV01_20191107_451324_G3 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451325_E7 (3%)","Jackson_Lab_John_MURGIGV01_20191107_451305_D11 (15%)","Jackson_Lab_John_MURGIGV01_20191107_451500_B11 (4%)","Jackson_Lab_John_MURGIGV01_20191107_451510_C2 (3%)","Jackson_Lab_John_MURGIGV01_20191107_455737_D10 (49%)","Jackson_Lab_John_MURGIGV01_20191107_451626_F7 (4%)","Jackson_Lab_John_MURGIGV01_20191107_451627_H4 (5%)","Jackson_Lab_John_MURGIGV01_20191107_455877_D2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_455882_A5 (3%)","Jackson_Lab_John_MURGIGV01_20191107_455865_F2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_455898_G9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459904_D5 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459913_B6 (4%)","Jackson_Lab_John_MURGIGV01_20191107_459933_B1 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459951_E3 (6%)","Jackson_Lab_John_MURGIGV01_20191107_459942_C4 (4%)","Jackson_Lab_John_MURGIGV01_20191107_459970_B2 (3%)","Jackson_Lab_John_MURGIGV01_20191107_459961_A12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461047_B12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461035_A2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461043_C7 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461829_D11 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461832_D3 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461089_A11 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461814_C12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461819_H9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461878_G2 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461880_F12 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461867_D3 (4%)","Jackson_Lab_John_MURGIGV01_20191107_461889_E5 (3%)","Jackson_Lab_John_MURGIGV01_20191107_461906_A8 (3%)","Jackson_Lab_John_MURGIGV01_20191107_450631_C9 (3%)","Jackson_Lab_John_MURGIGV01_20191107_450652_C5 (3%)","Jackson_Lab_John_MURGIGV01_20191114_449206_D7 (4%)","Jackson_Lab_John_MURGIGV01_20191114_449262_E7 (30%)","Jackson_Lab_John_MURGIGV01_20191114_451160_C11 (4%)","Jackson_Lab_John_MURGIGV01_20191114_459939_H4 (3%)","Columbia_University_John_MURGIGV01_20200529_449250_G1 (3%)","Columbia_University_John_MURGIGV01_20200529_449260_D11 (3%)","Columbia_University_John_MURGIGV01_20200529_449262_D1 (3%)","Columbia_University_John_MURGIGV01_20200529_449271_H1 (3%)","Columbia_University_John_MURGIGV01_20200529_449290_G11 (3%)","Columbia_University_John_MURGIGV01_20200529_450416_B8 (3%)","Columbia_University_John_MURGIGV01_20200529_450428_B5 (3%)","Columbia_University_John_MURGIGV01_20200529_450411_D3 (3%)","Columbia_University_John_MURGIGV01_20200529_450412_B4 (3%)","Columbia_University_John_MURGIGV01_20200529_451140_B12 (3%)","Columbia_University_John_MURGIGV01_20200529_455737_D5 (3%)","Columbia_University_John_MURGIGV01_20200529_451704_F9 (3%)","Columbia_University_John_MURGIGV01_20200529_451705_C12 (3%)","Columbia_University_John_MURGIGV01_20200529_451707_G11 (3%)","Columbia_University_John_MURGIGV01_20200529_451708_F10 (3%)","Columbia_University_John_MURGIGV01_20200529_461865_A5 (3%)","Columbia_University_John_MURGIGV01_20200529_451825_F5 (3%)","Columbia_University_John_MURGIGV01_20200529_449183_C8 (3%)","Columbia_University_John_MURGIGV01_20200529_449281_H8 (3%)","Columbia_University_John_MURGIGV01_20200529_451290_F12 (5%)","Columbia_University_John_MURGIGV01_20200529_451305_A1 (4%)","Columbia_University_John_MURGIGV01_20200529_451706_D4 (3%)"]},"chartOpts":{"pointcolor":["#2ecc40","#b10dc9","#ff4136"],"pointsize":4,"xlab":"Average X chr intensity","ylab":"Proportion het on X chr"}},"evals":[],"jsHooks":[]}</script>
<p>In the above scatterplot, we show the proportion of hets vs the average intensity for the X chromosome SNPs. In calculating the proportion of heterozygous genotypes for the individuals, we look at X chromosome genotypes equal to 2 which corresponds to the heterozygote) relative to not being 0 (which is used to encode missing genotypes). The genotypes are arranged with rows being individuals and columns being markers.</p>
<p>The following are the mice that have had sex incorrectly assigned:</p>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
Unique.Sample.ID
</th>
<th style="text-align:center;">
Sex
</th>
<th style="text-align:center;">
Inferred.Sex
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
449250
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449250_B4
</td>
<td style="text-align:center;">
M
</td>
<td style="text-align:center;">
F
</td>
</tr>
<tr>
<td style="text-align:center;">
450412
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450412_A8
</td>
<td style="text-align:center;">
M
</td>
<td style="text-align:center;">
F
</td>
</tr>
<tr>
<td style="text-align:center;">
461865
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_461865_F7
</td>
<td style="text-align:center;">
M
</td>
<td style="text-align:center;">
F
</td>
</tr>
<tr>
<td style="text-align:center;">
449183
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449183_C10
</td>
<td style="text-align:center;">
M
</td>
<td style="text-align:center;">
F
</td>
</tr>
<tr>
<td style="text-align:center;">
449281
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:center;">
M
</td>
<td style="text-align:center;">
F
</td>
</tr>
<tr>
<td style="text-align:center;">
450411
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_450411_H12
</td>
<td style="text-align:center;">
M
</td>
<td style="text-align:center;">
F
</td>
</tr>
</tbody>
</table>
</div>
<div id="sample-duplicates" class="section level3">
<h3>Sample Duplicates</h3>
<pre class="r"><code>cg &lt;- compare_geno(gm, cores=10)
summary.cg &lt;- summary(cg)</code></pre>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
ind1
</th>
<th style="text-align:left;">
iid1
</th>
<th style="text-align:center;">
ind2
</th>
<th style="text-align:center;">
iid2
</th>
<th style="text-align:center;">
prop_match
</th>
<th style="text-align:center;">
n_mismatch
</th>
<th style="text-align:center;">
n_typed
</th>
<th style="text-align:center;">
n_match
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191031_448182_B5
</td>
<td style="text-align:left;">
448182
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_448181_B3
</td>
<td style="text-align:center;">
448181
</td>
<td style="text-align:center;">
0.9984728
</td>
<td style="text-align:center;">
202
</td>
<td style="text-align:center;">
132268
</td>
<td style="text-align:center;">
132066
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191107_451289_C8
</td>
<td style="text-align:left;">
451289
</td>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_451290_F12
</td>
<td style="text-align:center;">
451290
</td>
<td style="text-align:center;">
0.9962084
</td>
<td style="text-align:center;">
494
</td>
<td style="text-align:center;">
130288
</td>
<td style="text-align:center;">
129794
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:left;">
449281
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_451305_D11
</td>
<td style="text-align:center;">
451305
</td>
<td style="text-align:center;">
0.9858624
</td>
<td style="text-align:center;">
1440
</td>
<td style="text-align:center;">
101856
</td>
<td style="text-align:center;">
100416
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191031_451140_E11
</td>
<td style="text-align:left;">
451140
</td>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_451140_B12
</td>
<td style="text-align:center;">
451140
</td>
<td style="text-align:center;">
0.9770514
</td>
<td style="text-align:center;">
2748
</td>
<td style="text-align:center;">
119746
</td>
<td style="text-align:center;">
116998
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191031_450416_D10
</td>
<td style="text-align:left;">
450416
</td>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_450416_B8
</td>
<td style="text-align:center;">
450416
</td>
<td style="text-align:center;">
0.9640625
</td>
<td style="text-align:center;">
4221
</td>
<td style="text-align:center;">
117454
</td>
<td style="text-align:center;">
113233
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191031_449260_G7
</td>
<td style="text-align:left;">
449260
</td>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_449260_D11
</td>
<td style="text-align:center;">
449260
</td>
<td style="text-align:center;">
0.9342110
</td>
<td style="text-align:center;">
7581
</td>
<td style="text-align:center;">
115232
</td>
<td style="text-align:center;">
107651
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191031_461865_F7
</td>
<td style="text-align:left;">
461865
</td>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_461865_A5
</td>
<td style="text-align:center;">
461865
</td>
<td style="text-align:center;">
0.9319680
</td>
<td style="text-align:center;">
7544
</td>
<td style="text-align:center;">
110889
</td>
<td style="text-align:center;">
103345
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191107_449183_C10
</td>
<td style="text-align:left;">
449183
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_455737_D10
</td>
<td style="text-align:center;">
455737
</td>
<td style="text-align:center;">
0.9126572
</td>
<td style="text-align:center;">
3431
</td>
<td style="text-align:center;">
39282
</td>
<td style="text-align:center;">
35851
</td>
</tr>
<tr>
<td style="text-align:left;">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:left;">
449281
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_455737_D10
</td>
<td style="text-align:center;">
455737
</td>
<td style="text-align:center;">
0.9002503
</td>
<td style="text-align:center;">
6258
</td>
<td style="text-align:center;">
62737
</td>
<td style="text-align:center;">
56479
</td>
</tr>
</tbody>
</table>
<p>Here is a histogram of the proportion of matching genotypes. The tick marks below the histogram indicate individual pairs.</p>
<pre class="r"><code>save(summary.cg,file = &quot;data/summary.cg_b6.v265d.RData&quot;)

#pdf(file = &quot;output/Proportion_matching_genotypes_before_removal_of_bad_samples.pdf&quot;, width = 20, height = 20) 
#par(mar=c(5.1,0.6,0.6, 0.6))
#hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
#     main=&quot;&quot;, yaxt=&quot;n&quot;, ylab=&quot;&quot;, xlab=&quot;Proportion matching genotypes&quot;)
#rug(cg[upper.tri(cg)])
#dev.off()

par(mar=c(5.1,0.6,0.6, 0.6))
hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
     main=&quot;&quot;, yaxt=&quot;n&quot;, ylab=&quot;&quot;, xlab=&quot;Proportion matching genotypes&quot;)
rug(cg[upper.tri(cg)])</code></pre>
<p><img src="figure/2.1_sample_bqc_b6.v265d.Rmd/Sample%20duplicates%20figures-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Sample-duplicates-figures-1">
Past versions of “Sample duplicates figures-1.png”
</button>
</p>
<div id="fig-no-spaces-Sample-duplicates-figures-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/blob/575cd8cd86ca14e8e7917782b914120389cd82c1/docs/figure/2.1_sample_bqc_b6.v265d.Rmd/Sample duplicates figures-1.png" target="_blank">575cd8c</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The majority of pairs share about 50% genotypes, but there is a group of pairs that are more closely related and share about 70% genotypes. The pairs cited above have &gt;90% matching genotypes.</p>
</div>
<div id="array-intensities" class="section level3">
<h3>Array Intensities</h3>
<pre class="r"><code>#load the intensities.fst.RData
#load(&quot;data/intensities.fst.RData&quot;)

#X and Y channel
#X &lt;- result[result$channel == &quot;X&quot;,]
#X &lt;- x[chrx,ind_ids(gm)]
#rownames(X) &lt;- X$snp
#X &lt;- X[,c(-1,-2)]

#Y &lt;- result[result$channel == &quot;Y&quot;,]
#rownames(Y) &lt;- Y$snp
#Y &lt;- Y[,c(-1,-2)]

#int &lt;- result

#rm(result)
#int &lt;- int[seq(1, nrow(int), by=2),-(1:2)] + int[-seq(1, nrow(int), by=2),-(1:2)]
#int &lt;- int[,intersect(ind_ids(gm), colnames(int))]
#n &lt;- names(sort(percent_missing[intersect(ind_ids(gm), colnames(int))], decreasing=TRUE))
#iboxplot(log10(t(int[,n])+1), orderByMedian=FALSE, chartOpts=list(ylab=&quot;log10(SNP intensity + 1)&quot;))</code></pre>
<p>In the above plot, distributions of array intensities (after a log10(x+1) transformation) are displayed.</p>
<p>The arrays are sorted by the proportion of missing genotype data for the sample, and the curves connect various quantiles of the intensities.</p>
<pre class="r"><code>#qu &lt;- apply(int, 2, quantile, c(0.01, 0.99), na.rm=TRUE)
#group &lt;- (percent_missing &gt;= 19.97) + (percent_missing &gt; 5) + (percent_missing &gt; 2) + 1
#labels &lt;- paste0(colnames(qu), &quot; (&quot;, round(percent_missing), &quot;%)&quot;)
#iplot(qu[1,], qu[2,], indID=labels, group=group,
#      chartOpts=list(xlab=&quot;1 %ile of array intensities&quot;,
#                     ylab=&quot;99 %ile of array intensities&quot;,
#                     pointcolor=c(&quot;#ccc&quot;, &quot;slateblue&quot;, &quot;Orchid&quot;, &quot;#ff851b&quot;)))</code></pre>
<p>For this particular set of arrays, a plot of the 1 %ile vs the 99 %ile is quite revealing. In the following, the orange points are those with &gt; 20% missing genotypes, the pink points are the samples with 5-20% missing genotypes, and the blue points are the samples with 2-5% missing genotypes.</p>
</div>
<div id="genotype-frequencies" class="section level3">
<h3>Genotype Frequencies</h3>
<p>The following triangle plots show the genotype frequency distributions for the mice, among the four groups of markers with common minor allele frequency (MAF) in the founder strains. These plots make use of the fact that for a point within an equilateral triangle, the sum of the distances to the three sides is a constant.</p>
<pre class="r"><code>g &lt;- do.call(&quot;cbind&quot;, gm$geno[1:19])
fg &lt;- do.call(&quot;cbind&quot;, gm$founder_geno[1:19])
g &lt;- g[,colSums(fg==0)==0]
fg &lt;- fg[,colSums(fg==0)==0]
fgn &lt;- colSums(fg==3)

gf_ind &lt;- vector(&quot;list&quot;, 4)
for(i in 1:4) {
  gf_ind[[i]] &lt;- t(apply(g[,fgn==i], 1, function(a) table(factor(a, 1:3))/sum(a != 0)))
}

par(mfrow=c(4,1), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c(&quot;AA&quot;, &quot;AB&quot;, &quot;BB&quot;), main=paste0(&quot;MAF = &quot;, i, &quot;/8&quot;))
  tripoints(gf_ind[[i]], pch=21, bg=&quot;lightblue&quot;)
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg=&quot;violetred&quot;)
  
  if(i&gt;=3) { # label mouse with lowest het
    wh &lt;- which(gf_ind[[i]][,2] == min(gf_ind[[i]][,2]))
    tritext(gf_ind[[i]][wh,,drop=FALSE] + c(0.02, -0.02, 0),
            names(wh), adj=c(0, 1))
  }
  
  # label other mice
  if(i==1) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.3]
  }
  else if(i==2) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.48]
  }
  else if(i==3) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.51]
  }
  else if(i==4) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.6]
  }
  
  for(ind in lab) {
    if(grepl(&quot;^F&quot;, ind) &amp;&amp; i != 3) {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(-0.01, 0, +0.01), ind, adj=c(1,0.5))
    } else {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(0.01, 0, -0.01), ind, adj=c(0,0.5))
    }
  }
}</code></pre>
<p><img src="figure/2.1_sample_bqc_b6.v265d.Rmd/Genotype%20frequencies-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Genotype-frequencies-1">
Past versions of “Genotype frequencies-1.png”
</button>
</p>
<div id="fig-no-spaces-Genotype-frequencies-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/blob/575cd8cd86ca14e8e7917782b914120389cd82c1/docs/figure/2.1_sample_bqc_b6.v265d.Rmd/Genotype frequencies-1.png" target="_blank">575cd8c</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The majority of individuals are tightly clustered around the expected distribution (in pink). There are a few outliers with elevated &amp; reduced heterozygosity, but most of these are among the samples with highest rates of missing genotypes.</p>
</div>
<div id="crossover-counts" class="section level3">
<h3>Crossover Counts</h3>
<pre class="r"><code>#load pre-caluated results
#load(&quot;data/pr.RData&quot;)
pr&lt;-readRDS(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/haplo.reconst/2020-11-11/b6.v265d/all_samples_sorted/Jax.Columbia_John__GigaMUGA_genoprobs_36state_sorted.rds&quot;)
#load(&quot;data/m.RData&quot;)
#load(&quot;data/nxo.RData&quot;)

m &lt;- maxmarg(pr)
nxo &lt;- count_xo(m)
#save(pr, file = &quot;data/pr.RData&quot;)
#str(pr)
save(m, file = &quot;data/m_b6.v265d.RData&quot;)
save(nxo, file = &quot;data/nxo_b6.v265d.RData&quot;)

e &lt;- calc_errorlod(gm, pr)
#str(e)
e &lt;- do.call(&quot;cbind&quot;, e)
save(e, file = &quot;data/e_b6.v265d.RData&quot;)

#snpg &lt;- predict_snpgeno(gm, m)
#snpg &lt;- do.call(&quot;cbind&quot;, snpg)
#save(snpg, file = &quot;data/snpg_b6.v265d.RData&quot;)

#crossover
totxo &lt;- rowSums(nxo)[rownames(gm$covar)]
iplot(seq_along(totxo),
      totxo,
      group=gm$covar$generation,
      chartOpts=list(xlab=&quot;Mouse&quot;, ylab=&quot;Number of crossovers&quot;, 
                     margin=list(left=80,top=40,right=40,bottom=40,inner=5),
                     axispos=list(xtitle=25,ytitle=50,xlabel=5,ylabel=5)))</code></pre>
<div id="htmlwidget-2ff5ea4871f66ca79aa6" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-2ff5ea4871f66ca79aa6">{"x":{"data":{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202],"y":[307,333,313,319,289,299,308,313,304,341,311,273,290,326,333,343,329,294,294,343,322,301,2384,285,982,1336,322,286,329,501,318,2450,345,344,1466,316,309,322,300,303,330,316,317,334,306,301,344,304,295,304,323,337,1387,518,330,300,291,331,321,311,296,276,289,329,316,309,310,328,333,342,308,317,311,376,310,301,327,337,348,302,308,318,326,314,327,315,318,290,297,305,320,358,309,314,300,321,319,328,322,307,318,310,318,295,326,319,282,312,313,324,314,350,319,324,323,320,312,298,303,320,311,817,315,334,306,329,311,304,2101,342,313,317,2604,296,200,340,3857,298,318,316,298,297,324,328,58,304,313,1478,320,308,314,336,327,332,322,327,295,365,327,288,335,335,327,318,312,312,310,307,323,329,293,295,313,315,333,328,332,2204,318,332,308,272,284,326,296,310,312,302,299,1164,306,333,323,314,311,279,325,328,319,301,319,333],"group":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"indID":["Jackson_Lab_John_MURGIGV01_20190728_451442_A3","Jackson_Lab_John_MURGIGV01_20190728_459954_A7","Jackson_Lab_John_MURGIGV01_20190728_449298_A10","Jackson_Lab_John_MURGIGV01_20191031_450600_G6","Jackson_Lab_John_MURGIGV01_20191031_450738_B12","Jackson_Lab_John_MURGIGV01_20191031_450740_G9","Jackson_Lab_John_MURGIGV01_20191031_445490_G4","Jackson_Lab_John_MURGIGV01_20191031_445491_F2","Jackson_Lab_John_MURGIGV01_20191031_448191_D12","Jackson_Lab_John_MURGIGV01_20191031_448193_D11","Jackson_Lab_John_MURGIGV01_20191031_448182_B5","Jackson_Lab_John_MURGIGV01_20191031_448918_C5","Jackson_Lab_John_MURGIGV01_20191031_448919_A8","Jackson_Lab_John_MURGIGV01_20191031_448903_A5","Jackson_Lab_John_MURGIGV01_20191031_448905_A4","Jackson_Lab_John_MURGIGV01_20191031_449187_E4","Jackson_Lab_John_MURGIGV01_20191031_449189_G10","Jackson_Lab_John_MURGIGV01_20191031_449201_A10","Jackson_Lab_John_MURGIGV01_20191031_449203_F2","Jackson_Lab_John_MURGIGV01_20191031_449207_E2","Jackson_Lab_John_MURGIGV01_20191031_449211_H4","Jackson_Lab_John_MURGIGV01_20191031_449248_A1","Jackson_Lab_John_MURGIGV01_20191031_449250_B4","Jackson_Lab_John_MURGIGV01_20191031_449258_A9","Jackson_Lab_John_MURGIGV01_20191031_449260_G7","Jackson_Lab_John_MURGIGV01_20191031_449271_G4","Jackson_Lab_John_MURGIGV01_20191031_449273_D10","Jackson_Lab_John_MURGIGV01_20191031_449284_B9","Jackson_Lab_John_MURGIGV01_20191031_449299_D7","Jackson_Lab_John_MURGIGV01_20191031_450416_D10","Jackson_Lab_John_MURGIGV01_20191031_449283_F5","Jackson_Lab_John_MURGIGV01_20191031_450428_E12","Jackson_Lab_John_MURGIGV01_20191031_450433_G1","Jackson_Lab_John_MURGIGV01_20191031_450434_D2","Jackson_Lab_John_MURGIGV01_20191031_450412_A8","Jackson_Lab_John_MURGIGV01_20191031_451041_F10","Jackson_Lab_John_MURGIGV01_20191031_451043_A6","Jackson_Lab_John_MURGIGV01_20191031_451049_C9","Jackson_Lab_John_MURGIGV01_20191031_451050_C12","Jackson_Lab_John_MURGIGV01_20191031_451052_C5","Jackson_Lab_John_MURGIGV01_20191031_451053_F9","Jackson_Lab_John_MURGIGV01_20191031_451066_H10","Jackson_Lab_John_MURGIGV01_20191031_451067_E1","Jackson_Lab_John_MURGIGV01_20191031_451071_H12","Jackson_Lab_John_MURGIGV01_20191031_451153_H1","Jackson_Lab_John_MURGIGV01_20191031_451155_E6","Jackson_Lab_John_MURGIGV01_20191031_451161_F12","Jackson_Lab_John_MURGIGV01_20191031_451183_F10","Jackson_Lab_John_MURGIGV01_20191031_451186_D8","Jackson_Lab_John_MURGIGV01_20191031_451200_D7","Jackson_Lab_John_MURGIGV01_20191031_451163_B1","Jackson_Lab_John_MURGIGV01_20191031_451165_G11","Jackson_Lab_John_MURGIGV01_20191031_451140_E11","Jackson_Lab_John_MURGIGV01_20191031_451290_D1","Jackson_Lab_John_MURGIGV01_20191031_451295_H4","Jackson_Lab_John_MURGIGV01_20191031_451298_E5","Jackson_Lab_John_MURGIGV01_20191031_451319_D9","Jackson_Lab_John_MURGIGV01_20191031_451336_B8","Jackson_Lab_John_MURGIGV01_20191031_451337_G8","Jackson_Lab_John_MURGIGV01_20191031_451302_H6","Jackson_Lab_John_MURGIGV01_20191031_454536_D6","Jackson_Lab_John_MURGIGV01_20191031_454537_H9","Jackson_Lab_John_MURGIGV01_20191031_451435_G3","Jackson_Lab_John_MURGIGV01_20191031_451441_F3","Jackson_Lab_John_MURGIGV01_20191031_451444_C7","Jackson_Lab_John_MURGIGV01_20191031_451445_F5","Jackson_Lab_John_MURGIGV01_20191031_451460_C10","Jackson_Lab_John_MURGIGV01_20191031_451461_E7","Jackson_Lab_John_MURGIGV01_20191031_451464_C7","Jackson_Lab_John_MURGIGV01_20191031_451466_C4","Jackson_Lab_John_MURGIGV01_20191031_451473_F11","Jackson_Lab_John_MURGIGV01_20191031_451475_D3","Jackson_Lab_John_MURGIGV01_20191031_451508_B6","Jackson_Lab_John_MURGIGV01_20191031_451518_D7","Jackson_Lab_John_MURGIGV01_20191031_451519_C12","Jackson_Lab_John_MURGIGV01_20191031_451560_D4","Jackson_Lab_John_MURGIGV01_20191031_451543_D1","Jackson_Lab_John_MURGIGV01_20191031_451551_E12","Jackson_Lab_John_MURGIGV01_20191031_451552_H6","Jackson_Lab_John_MURGIGV01_20191031_451556_H7","Jackson_Lab_John_MURGIGV01_20191031_451557_B9","Jackson_Lab_John_MURGIGV01_20191031_451566_E7","Jackson_Lab_John_MURGIGV01_20191031_451568_H11","Jackson_Lab_John_MURGIGV01_20191031_455738_G5","Jackson_Lab_John_MURGIGV01_20191031_451577_H2","Jackson_Lab_John_MURGIGV01_20191031_451579_C5","Jackson_Lab_John_MURGIGV01_20191031_451632_B3","Jackson_Lab_John_MURGIGV01_20191031_451649_A11","Jackson_Lab_John_MURGIGV01_20191031_451655_E1","Jackson_Lab_John_MURGIGV01_20191031_451656_D4","Jackson_Lab_John_MURGIGV01_20191031_455874_D8","Jackson_Lab_John_MURGIGV01_20191031_455881_G8","Jackson_Lab_John_MURGIGV01_20191031_455864_F2","Jackson_Lab_John_MURGIGV01_20191031_459907_E6","Jackson_Lab_John_MURGIGV01_20191031_459909_B8","Jackson_Lab_John_MURGIGV01_20191031_455899_G11","Jackson_Lab_John_MURGIGV01_20191031_459901_A4","Jackson_Lab_John_MURGIGV01_20191031_459915_H2","Jackson_Lab_John_MURGIGV01_20191031_459922_B4","Jackson_Lab_John_MURGIGV01_20191031_459923_E9","Jackson_Lab_John_MURGIGV01_20191031_459934_A10","Jackson_Lab_John_MURGIGV01_20191031_459936_G9","Jackson_Lab_John_MURGIGV01_20191031_459943_C11","Jackson_Lab_John_MURGIGV01_20191031_459969_A4","Jackson_Lab_John_MURGIGV01_20191031_459959_G2","Jackson_Lab_John_MURGIGV01_20191031_461048_B11","Jackson_Lab_John_MURGIGV01_20191031_461036_A7","Jackson_Lab_John_MURGIGV01_20191031_461042_G5","Jackson_Lab_John_MURGIGV01_20191031_461056_C6","Jackson_Lab_John_MURGIGV01_20191031_461057_A9","Jackson_Lab_John_MURGIGV01_20191031_461808_A6","Jackson_Lab_John_MURGIGV01_20191031_461809_A12","Jackson_Lab_John_MURGIGV01_20191031_461088_G1","Jackson_Lab_John_MURGIGV01_20191031_461096_C3","Jackson_Lab_John_MURGIGV01_20191031_461098_C2","Jackson_Lab_John_MURGIGV01_20191031_461812_A5","Jackson_Lab_John_MURGIGV01_20191031_461818_C6","Jackson_Lab_John_MURGIGV01_20191031_461820_A7","Jackson_Lab_John_MURGIGV01_20191031_461821_C9","Jackson_Lab_John_MURGIGV01_20191031_461881_E8","Jackson_Lab_John_MURGIGV01_20191031_461884_B9","Jackson_Lab_John_MURGIGV01_20191031_461865_F7","Jackson_Lab_John_MURGIGV01_20191031_461866_F6","Jackson_Lab_John_MURGIGV01_20191031_461868_E8","Jackson_Lab_John_MURGIGV01_20191031_461888_H10","Jackson_Lab_John_MURGIGV01_20191107_450598_A1","Jackson_Lab_John_MURGIGV01_20191107_448181_B3","Jackson_Lab_John_MURGIGV01_20191107_448906_D7","Jackson_Lab_John_MURGIGV01_20191107_449183_C10","Jackson_Lab_John_MURGIGV01_20191107_449191_A10","Jackson_Lab_John_MURGIGV01_20191107_449192_C9","Jackson_Lab_John_MURGIGV01_20191107_449259_E4","Jackson_Lab_John_MURGIGV01_20191107_449290_G1","Jackson_Lab_John_MURGIGV01_20191107_450413_F10","Jackson_Lab_John_MURGIGV01_20191107_449281_C12","Jackson_Lab_John_MURGIGV01_20191107_450426_F5","Jackson_Lab_John_MURGIGV01_20191107_450411_H12","Jackson_Lab_John_MURGIGV01_20191107_451070_B7","Jackson_Lab_John_MURGIGV01_20191107_451170_F4","Jackson_Lab_John_MURGIGV01_20191107_451203_F11","Jackson_Lab_John_MURGIGV01_20191107_451289_C8","Jackson_Lab_John_MURGIGV01_20191107_451315_C7","Jackson_Lab_John_MURGIGV01_20191107_451324_G3","Jackson_Lab_John_MURGIGV01_20191107_451325_E7","Jackson_Lab_John_MURGIGV01_20191107_451305_D11","Jackson_Lab_John_MURGIGV01_20191107_451500_B11","Jackson_Lab_John_MURGIGV01_20191107_451510_C2","Jackson_Lab_John_MURGIGV01_20191107_455737_D10","Jackson_Lab_John_MURGIGV01_20191107_451626_F7","Jackson_Lab_John_MURGIGV01_20191107_451627_H4","Jackson_Lab_John_MURGIGV01_20191107_455877_D2","Jackson_Lab_John_MURGIGV01_20191107_455882_A5","Jackson_Lab_John_MURGIGV01_20191107_455865_F2","Jackson_Lab_John_MURGIGV01_20191107_455898_G9","Jackson_Lab_John_MURGIGV01_20191107_459904_D5","Jackson_Lab_John_MURGIGV01_20191107_459913_B6","Jackson_Lab_John_MURGIGV01_20191107_459933_B1","Jackson_Lab_John_MURGIGV01_20191107_459951_E3","Jackson_Lab_John_MURGIGV01_20191107_459942_C4","Jackson_Lab_John_MURGIGV01_20191107_459970_B2","Jackson_Lab_John_MURGIGV01_20191107_459961_A12","Jackson_Lab_John_MURGIGV01_20191107_461047_B12","Jackson_Lab_John_MURGIGV01_20191107_461035_A2","Jackson_Lab_John_MURGIGV01_20191107_461043_C7","Jackson_Lab_John_MURGIGV01_20191107_461829_D11","Jackson_Lab_John_MURGIGV01_20191107_461832_D3","Jackson_Lab_John_MURGIGV01_20191107_461089_A11","Jackson_Lab_John_MURGIGV01_20191107_461814_C12","Jackson_Lab_John_MURGIGV01_20191107_461819_H9","Jackson_Lab_John_MURGIGV01_20191107_461878_G2","Jackson_Lab_John_MURGIGV01_20191107_461880_F12","Jackson_Lab_John_MURGIGV01_20191107_461867_D3","Jackson_Lab_John_MURGIGV01_20191107_461889_E5","Jackson_Lab_John_MURGIGV01_20191107_461906_A8","Jackson_Lab_John_MURGIGV01_20191107_450631_C9","Jackson_Lab_John_MURGIGV01_20191107_450652_C5","Jackson_Lab_John_MURGIGV01_20191114_449206_D7","Jackson_Lab_John_MURGIGV01_20191114_449262_E7","Jackson_Lab_John_MURGIGV01_20191114_451160_C11","Jackson_Lab_John_MURGIGV01_20191114_459939_H4","Columbia_University_John_MURGIGV01_20200529_449250_G1","Columbia_University_John_MURGIGV01_20200529_449260_D11","Columbia_University_John_MURGIGV01_20200529_449262_D1","Columbia_University_John_MURGIGV01_20200529_449271_H1","Columbia_University_John_MURGIGV01_20200529_449290_G11","Columbia_University_John_MURGIGV01_20200529_450416_B8","Columbia_University_John_MURGIGV01_20200529_450428_B5","Columbia_University_John_MURGIGV01_20200529_450411_D3","Columbia_University_John_MURGIGV01_20200529_450412_B4","Columbia_University_John_MURGIGV01_20200529_451140_B12","Columbia_University_John_MURGIGV01_20200529_455737_D5","Columbia_University_John_MURGIGV01_20200529_451704_F9","Columbia_University_John_MURGIGV01_20200529_451705_C12","Columbia_University_John_MURGIGV01_20200529_451707_G11","Columbia_University_John_MURGIGV01_20200529_451708_F10","Columbia_University_John_MURGIGV01_20200529_461865_A5","Columbia_University_John_MURGIGV01_20200529_451825_F5","Columbia_University_John_MURGIGV01_20200529_449183_C8","Columbia_University_John_MURGIGV01_20200529_449281_H8","Columbia_University_John_MURGIGV01_20200529_451290_F12","Columbia_University_John_MURGIGV01_20200529_451305_A1","Columbia_University_John_MURGIGV01_20200529_451706_D4"]},"chartOpts":{"xlab":"Mouse","ylab":"Number of crossovers","margin":{"left":80,"top":40,"right":40,"bottom":40,"inner":5},"axispos":{"xtitle":25,"ytitle":50,"xlabel":5,"ylabel":5}}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>#save crossover into pdf
pdf(file = &quot;output/number_crossover_b6.v265d.pdf&quot;)
cross_over &lt;- data.frame(Mouse = seq_along(totxo), Number_crossovers = totxo, generation = gm$covar$generation)
names(totxo) &lt;- as.character(do.call(rbind.data.frame, strsplit(names(totxo), &quot;V01_&quot;))[,2])
names(totxo)[totxo &gt;= 200 &amp; totxo &lt;= 800] = &quot;&quot;
# Change point shapes and colors
p &lt;-ggplot(cross_over, aes(x=Mouse, y=Number_crossovers, fill = generation, color=generation)) +
  geom_point() +
  geom_text_repel(aes(label=names(totxo),hjust=0,vjust=0), show.legend = FALSE)
p
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="figure/2.1_sample_bqc_b6.v265d.Rmd/Crossover%20counts-2.png" width="912" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Crossover-counts-2">
Past versions of “Crossover counts-2.png”
</button>
</p>
<div id="fig-no-spaces-Crossover-counts-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheJacksonLaboratory/John-Tolman_Workflow/blob/575cd8cd86ca14e8e7917782b914120389cd82c1/docs/figure/2.1_sample_bqc_b6.v265d.Rmd/Crossover counts-2.png" target="_blank">575cd8c</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2020-11-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Counts of inferred crossovers can be a useful diagnostic for identifying problem samples, which may show an excessive number of apparent crossovers. Above is a plot of the number of crossovers vs the mouse index, colored by generation.</p>
<p>We’re looking for mice with an excessive number of crossovers as well as mice with very few crossovers. These are shown below</p>
<p><strong>NB:</strong> DO generation of an individual is taken into account.</p>
<pre><code>                                               percent_missing total_xo
Jackson_Lab_John_MURGIGV01_20191031_459934_A10            5.11      318
Jackson_Lab_John_MURGIGV01_20191031_449284_B9             5.22      286
Jackson_Lab_John_MURGIGV01_20191031_455881_G8             5.36      358
Jackson_Lab_John_MURGIGV01_20191031_461809_A12            6.03      350
Jackson_Lab_John_MURGIGV01_20191031_451071_H12            6.20      334
Jackson_Lab_John_MURGIGV01_20191107_459951_E3             6.39      365
Jackson_Lab_John_MURGIGV01_20191031_451518_D7             7.29      376
Jackson_Lab_John_MURGIGV01_20191031_451290_D1            10.82      518
Jackson_Lab_John_MURGIGV01_20191031_451140_E11           12.36     1387
Jackson_Lab_John_MURGIGV01_20191031_450416_D10           14.19      501
Jackson_Lab_John_MURGIGV01_20191107_451305_D11           14.61       58
Jackson_Lab_John_MURGIGV01_20191031_449260_G7            15.72      982
Jackson_Lab_John_MURGIGV01_20191031_449271_G4            17.47     1336
Jackson_Lab_John_MURGIGV01_20191031_461865_F7            18.94      817
Jackson_Lab_John_MURGIGV01_20191107_449281_C12           20.38      200
Jackson_Lab_John_MURGIGV01_20191031_450412_A8            25.80     1466
Jackson_Lab_John_MURGIGV01_20191114_449262_E7            29.71     2204
Jackson_Lab_John_MURGIGV01_20191031_449250_B4            30.04     2384
Jackson_Lab_John_MURGIGV01_20191031_450428_E12           30.45     2450
Jackson_Lab_John_MURGIGV01_20191107_449290_G1            30.85     2604
Jackson_Lab_John_MURGIGV01_20191107_450411_H12           36.50     3857
Jackson_Lab_John_MURGIGV01_20191107_455737_D10           49.47     1478
Jackson_Lab_John_MURGIGV01_20191107_449183_C10           65.00     2101</code></pre>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
id
</th>
<th style="text-align:center;">
number.xo
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
449250
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449250_B4
</td>
<td style="text-align:center;">
2384
</td>
</tr>
<tr>
<td style="text-align:center;">
449271
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449271_G4
</td>
<td style="text-align:center;">
1336
</td>
</tr>
<tr>
<td style="text-align:center;">
450428
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450428_E12
</td>
<td style="text-align:center;">
2450
</td>
</tr>
<tr>
<td style="text-align:center;">
450412
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450412_A8
</td>
<td style="text-align:center;">
1466
</td>
</tr>
<tr>
<td style="text-align:center;">
451140
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451140_E11
</td>
<td style="text-align:center;">
1387
</td>
</tr>
<tr>
<td style="text-align:center;">
449183
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449183_C10
</td>
<td style="text-align:center;">
2101
</td>
</tr>
<tr>
<td style="text-align:center;">
449290
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449290_G1
</td>
<td style="text-align:center;">
2604
</td>
</tr>
<tr>
<td style="text-align:center;">
449281
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:center;">
200
</td>
</tr>
<tr>
<td style="text-align:center;">
450411
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_450411_H12
</td>
<td style="text-align:center;">
3857
</td>
</tr>
<tr>
<td style="text-align:center;">
451305
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_451305_D11
</td>
<td style="text-align:center;">
58
</td>
</tr>
<tr>
<td style="text-align:center;">
455737
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_455737_D10
</td>
<td style="text-align:center;">
1478
</td>
</tr>
<tr>
<td style="text-align:center;">
449262
</td>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191114_449262_E7
</td>
<td style="text-align:center;">
2204
</td>
</tr>
<tr>
<td style="text-align:center;">
451140
</td>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_451140_B12
</td>
<td style="text-align:center;">
1164
</td>
</tr>
</tbody>
</table>
</div>
<div id="genotyping-error-lod-scores" class="section level3">
<h3>Genotyping Error LOD Scores</h3>
<pre class="r"><code>#load(&quot;data/e_b6.v265d.RData&quot;)
errors_ind &lt;- rowSums(e&gt;2)[rownames(gm$covar)]/n_typed(gm)*100
lab &lt;- paste0(names(errors_ind), &quot; (&quot;, myround(percent_missing[rownames(gm$covar)],1), &quot;%)&quot;)
iplot(seq_along(errors_ind), errors_ind, indID=lab,
      chartOpts=list(xlab=&quot;Mouse&quot;, ylab=&quot;Percent genotyping errors&quot;, ylim=c(0, 8),
                     axispos=list(xtitle=25, ytitle=50, xlabel=5, ylabel=5)))</code></pre>
<div id="htmlwidget-563103dffe6d1a1e47ac" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-563103dffe6d1a1e47ac">{"x":{"data":{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202],"y":[0.60905,0.92512,0.59217,0.69282,0.084497,0.038333,0.62908,0.4305,0.060389,0.03768,0.040604,0.65814,1.2448,0.97179,0.5709,0.45879,0.44168,0.76623,0.42949,0.046045,0.035338,0.56712,3.8642,0.49822,0.42604,0.71831,0.0452,0.11373,0.73217,0.99556,0.6614,2.7822,0.044417,0.032327,2.0672,0.041351,0.05055,0.033082,0.060421,0.04437,0.040578,0.76216,0.48007,0.26322,0.72926,0.64613,0.54447,0.72542,0.63591,0.76804,0.04743,0.037621,10.168,1.6431,0.031603,0.034616,0.71753,0.41441,0.6491,0.064163,0.36644,0.67112,0.053516,0.65669,0.040601,0.046653,0.64168,0.34069,0.041389,0.046622,0.8528,0.96492,0.67499,0.64182,0.059751,0.95484,0.044375,1.0473,0.71951,1.0381,1.017,0.043747,0.049072,0.53861,0.061061,0.03839,0.50781,0.41686,0.48571,0.44531,0.049719,0.060794,0.036113,0.046655,0.039109,0.64282,0.59087,0.042198,0.039112,0.038337,1.2188,0.036832,0.032347,1.1371,0.9176,0.041419,0.70569,0.4434,0.048949,0.042196,0.55003,0.66498,0.050517,0.049632,0.032309,0.72551,0.8667,0.054957,0.045182,0.02933,0.30668,2.0916,0.68985,0.093101,0.036851,0.98256,0.055039,0.046038,37.04,0.058878,0.052814,0.70662,4.0785,0.069,49.016,0.04907,5.1832,0.055828,0.044522,0.73459,0.63294,0.52537,0.051292,0.044529,49.58,0.04756,0.68256,44.988,0.51472,0.53549,0.4772,0.056541,0.043023,0.64278,0.76354,0.079526,1.5637,0.36102,0.073612,0.62321,0.92071,0.046837,0.7858,0.45129,0.062654,0.049738,0.048978,0.60548,0.97835,0.49601,0.58693,0.062087,0.038446,0.055058,0.2699,0.7424,0.12067,3.8423,0.77642,0.061043,0.57059,0.036867,0.040653,0.038371,0.033087,0.029318,0.029381,0.47307,0.45516,12.132,0.58674,0.037632,0.035393,0.49316,0.77421,0.5581,0.84658,0.99928,0.58904,0.66057,0.15069,0.034593],"group":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"indID":["Jackson_Lab_John_MURGIGV01_20190728_451442_A3 (3.5%)","Jackson_Lab_John_MURGIGV01_20190728_459954_A7 (3.5%)","Jackson_Lab_John_MURGIGV01_20190728_449298_A10 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_450600_G6 (4.3%)","Jackson_Lab_John_MURGIGV01_20191031_450738_B12 (4.3%)","Jackson_Lab_John_MURGIGV01_20191031_450740_G9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_445490_G4 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_445491_F2 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_448191_D12 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_448193_D11 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_448182_B5 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_448918_C5 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_448919_A8 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_448903_A5 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_448905_A4 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_449187_E4 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_449189_G10 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_449201_A10 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_449203_F2 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_449207_E2 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_449211_H4 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_449248_A1 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_449250_B4 (30.0%)","Jackson_Lab_John_MURGIGV01_20191031_449258_A9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_449260_G7 (15.7%)","Jackson_Lab_John_MURGIGV01_20191031_449271_G4 (17.5%)","Jackson_Lab_John_MURGIGV01_20191031_449273_D10 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_449284_B9 (5.2%)","Jackson_Lab_John_MURGIGV01_20191031_449299_D7 (4.4%)","Jackson_Lab_John_MURGIGV01_20191031_450416_D10 (14.2%)","Jackson_Lab_John_MURGIGV01_20191031_449283_F5 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_450428_E12 (30.4%)","Jackson_Lab_John_MURGIGV01_20191031_450433_G1 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_450434_D2 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_450412_A8 (25.8%)","Jackson_Lab_John_MURGIGV01_20191031_451041_F10 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451043_A6 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_451049_C9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451050_C12 (3.6%)","Jackson_Lab_John_MURGIGV01_20191031_451052_C5 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451053_F9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451066_H10 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_451067_E1 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451071_H12 (6.2%)","Jackson_Lab_John_MURGIGV01_20191031_451153_H1 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451155_E6 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_451161_F12 (4.6%)","Jackson_Lab_John_MURGIGV01_20191031_451183_F10 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_451186_D8 (4.1%)","Jackson_Lab_John_MURGIGV01_20191031_451200_D7 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451163_B1 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_451165_G11 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451140_E11 (12.4%)","Jackson_Lab_John_MURGIGV01_20191031_451290_D1 (10.8%)","Jackson_Lab_John_MURGIGV01_20191031_451295_H4 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451298_E5 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451319_D9 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451336_B8 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_451337_G8 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451302_H6 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_454536_D6 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_454537_H9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451435_G3 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_451441_F3 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451444_C7 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451445_F5 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451460_C10 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451461_E7 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451464_C7 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451466_C4 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451473_F11 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451475_D3 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451508_B6 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_451518_D7 (7.3%)","Jackson_Lab_John_MURGIGV01_20191031_451519_C12 (3.7%)","Jackson_Lab_John_MURGIGV01_20191031_451560_D4 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_451543_D1 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451551_E12 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_451552_H6 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_451556_H7 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_451557_B9 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451566_E7 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_451568_H11 (3.5%)","Jackson_Lab_John_MURGIGV01_20191031_455738_G5 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451577_H2 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_451579_C5 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451632_B3 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_451649_A11 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_451655_E1 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_451656_D4 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_455874_D8 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_455881_G8 (5.4%)","Jackson_Lab_John_MURGIGV01_20191031_455864_F2 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459907_E6 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459909_B8 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_455899_G11 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459901_A4 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_459915_H2 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_459922_B4 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459923_E9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_459934_A10 (5.1%)","Jackson_Lab_John_MURGIGV01_20191031_459936_G9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_459943_C11 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459969_A4 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_459959_G2 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461048_B11 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461036_A7 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461042_G5 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_461056_C6 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461057_A9 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461808_A6 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461809_A12 (6.0%)","Jackson_Lab_John_MURGIGV01_20191031_461088_G1 (3.4%)","Jackson_Lab_John_MURGIGV01_20191031_461096_C3 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_461098_C2 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_461812_A5 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_461818_C6 (4.2%)","Jackson_Lab_John_MURGIGV01_20191031_461820_A7 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461821_C9 (3.3%)","Jackson_Lab_John_MURGIGV01_20191031_461881_E8 (3.2%)","Jackson_Lab_John_MURGIGV01_20191031_461884_B9 (3.1%)","Jackson_Lab_John_MURGIGV01_20191031_461865_F7 (18.9%)","Jackson_Lab_John_MURGIGV01_20191031_461866_F6 (3.9%)","Jackson_Lab_John_MURGIGV01_20191031_461868_E8 (3.8%)","Jackson_Lab_John_MURGIGV01_20191031_461888_H10 (3.2%)","Jackson_Lab_John_MURGIGV01_20191107_450598_A1 (5.0%)","Jackson_Lab_John_MURGIGV01_20191107_448181_B3 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_448906_D7 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_449183_C10 (65.0%)","Jackson_Lab_John_MURGIGV01_20191107_449191_A10 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_449192_C9 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_449259_E4 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_449290_G1 (30.9%)","Jackson_Lab_John_MURGIGV01_20191107_450413_F10 (3.9%)","Jackson_Lab_John_MURGIGV01_20191107_449281_C12 (20.4%)","Jackson_Lab_John_MURGIGV01_20191107_450426_F5 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_450411_H12 (36.5%)","Jackson_Lab_John_MURGIGV01_20191107_451070_B7 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451170_F4 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451203_F11 (3.3%)","Jackson_Lab_John_MURGIGV01_20191107_451289_C8 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451315_C7 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451324_G3 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_451325_E7 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451305_D11 (14.6%)","Jackson_Lab_John_MURGIGV01_20191107_451500_B11 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_451510_C2 (3.3%)","Jackson_Lab_John_MURGIGV01_20191107_455737_D10 (49.5%)","Jackson_Lab_John_MURGIGV01_20191107_451626_F7 (3.6%)","Jackson_Lab_John_MURGIGV01_20191107_451627_H4 (4.5%)","Jackson_Lab_John_MURGIGV01_20191107_455877_D2 (3.8%)","Jackson_Lab_John_MURGIGV01_20191107_455882_A5 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_455865_F2 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_455898_G9 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_459904_D5 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_459913_B6 (3.8%)","Jackson_Lab_John_MURGIGV01_20191107_459933_B1 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_459951_E3 (6.4%)","Jackson_Lab_John_MURGIGV01_20191107_459942_C4 (4.0%)","Jackson_Lab_John_MURGIGV01_20191107_459970_B2 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_459961_A12 (3.9%)","Jackson_Lab_John_MURGIGV01_20191107_461047_B12 (3.6%)","Jackson_Lab_John_MURGIGV01_20191107_461035_A2 (3.9%)","Jackson_Lab_John_MURGIGV01_20191107_461043_C7 (3.7%)","Jackson_Lab_John_MURGIGV01_20191107_461829_D11 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_461832_D3 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_461089_A11 (3.3%)","Jackson_Lab_John_MURGIGV01_20191107_461814_C12 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_461819_H9 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_461878_G2 (3.5%)","Jackson_Lab_John_MURGIGV01_20191107_461880_F12 (4.5%)","Jackson_Lab_John_MURGIGV01_20191107_461867_D3 (3.8%)","Jackson_Lab_John_MURGIGV01_20191107_461889_E5 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_461906_A8 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_450631_C9 (3.4%)","Jackson_Lab_John_MURGIGV01_20191107_450652_C5 (3.4%)","Jackson_Lab_John_MURGIGV01_20191114_449206_D7 (4.0%)","Jackson_Lab_John_MURGIGV01_20191114_449262_E7 (29.7%)","Jackson_Lab_John_MURGIGV01_20191114_451160_C11 (3.9%)","Jackson_Lab_John_MURGIGV01_20191114_459939_H4 (3.4%)","Columbia_University_John_MURGIGV01_20200529_449250_G1 (3.2%)","Columbia_University_John_MURGIGV01_20200529_449260_D11 (3.2%)","Columbia_University_John_MURGIGV01_20200529_449262_D1 (3.3%)","Columbia_University_John_MURGIGV01_20200529_449271_H1 (3.2%)","Columbia_University_John_MURGIGV01_20200529_449290_G11 (3.1%)","Columbia_University_John_MURGIGV01_20200529_450416_B8 (3.1%)","Columbia_University_John_MURGIGV01_20200529_450428_B5 (3.3%)","Columbia_University_John_MURGIGV01_20200529_450411_D3 (3.2%)","Columbia_University_John_MURGIGV01_20200529_450412_B4 (3.2%)","Columbia_University_John_MURGIGV01_20200529_451140_B12 (3.4%)","Columbia_University_John_MURGIGV01_20200529_455737_D5 (3.2%)","Columbia_University_John_MURGIGV01_20200529_451704_F9 (3.2%)","Columbia_University_John_MURGIGV01_20200529_451705_C12 (3.3%)","Columbia_University_John_MURGIGV01_20200529_451707_G11 (3.3%)","Columbia_University_John_MURGIGV01_20200529_451708_F10 (3.2%)","Columbia_University_John_MURGIGV01_20200529_461865_A5 (3.2%)","Columbia_University_John_MURGIGV01_20200529_451825_F5 (3.1%)","Columbia_University_John_MURGIGV01_20200529_449183_C8 (3.3%)","Columbia_University_John_MURGIGV01_20200529_449281_H8 (3.4%)","Columbia_University_John_MURGIGV01_20200529_451290_F12 (4.6%)","Columbia_University_John_MURGIGV01_20200529_451305_A1 (4.3%)","Columbia_University_John_MURGIGV01_20200529_451706_D4 (3.2%)"]},"chartOpts":{"xlab":"Mouse","ylab":"Percent genotyping errors","ylim":[0,8],"axispos":{"xtitle":25,"ytitle":50,"xlabel":5,"ylabel":5}}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>save(errors_ind, file = &quot;data/errors_ind.RData&quot;)</code></pre>
</div>
<div id="apparent-genotyping-errors" class="section level3">
<h3>Apparent Genotyping Errors</h3>
<pre class="r"><code>#load(&quot;data/snpg_b6.v265d.RData&quot;)
#snpg &lt;- gm$geno

#snpgn &lt;- snpg[rownames(gm$covar),]
snpgn &lt;- do.call(&quot;cbind&quot;, gm$geno)

gobs &lt;- do.call(&quot;cbind&quot;, gm$geno)
gobs[gobs==0] &lt;- NA

par(pty=&quot;s&quot;)
err_direct &lt;- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 &lt;- rowSums(e &gt; 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab=&quot;Percent errors (error LOD &gt; 0)&quot;,
         ylab=&quot;Percent errors (obs vs predicted)&quot;,
         xlim=c(0, 2), ylim=c(0, 2))
abline(0,1,lty=2, col=&quot;gray60&quot;)</code></pre>
<p><img src="figure/2.1_sample_bqc_b6.v265d.Rmd/apparent%20genotyping%20errors-1.png" width="912" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(file = &quot;output/Percent_genotype_errors_obs_vs_predicted_b6.v265d.pdf&quot;,width = 20, height = 20) 
par(pty=&quot;s&quot;)
err_direct &lt;- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 &lt;- rowSums(e &gt; 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab=&quot;Percent errors (error LOD &gt; 0)&quot;,
         ylab=&quot;Percent errors (obs vs predicted)&quot;,
         xlim=c(0, 2), ylim=c(0, 2))
abline(0,1,lty=2, col=&quot;gray60&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>par(pty=&quot;s&quot;)
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab=&quot;Percent errors (error LOD &gt; 0)&quot;,
         ylab=&quot;Percent errors (obs vs predicted)&quot;,
         xlim=c(0, 0.4), ylim=c(0, 0.4))
abline(0,1,lty=2, col=&quot;gray60&quot;)</code></pre>
<p><img src="figure/2.1_sample_bqc_b6.v265d.Rmd/apparent%20genotyping%20errors-2.png" width="912" style="display: block; margin: auto;" /></p>
<p>There were no samples with genotype errors greater than 1%</p>
</div>
<div id="removing-samples" class="section level3">
<h3>Removing Samples</h3>
<pre class="r"><code>##percent missing
gm.covar = data.frame(id=rownames(gm$covar),gm$covar)
qc_info &lt;- merge(gm.covar,
                  data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = &quot;id&quot;)

#cross_over
qc_info &lt;- merge(qc_info,
                 data.frame(id = rownames(cross_over),
                            Number_crossovers = cross_over$Number_crossovers,stringsAsFactors = F),by = &quot;id&quot;)

#missing sex
#qc_info$sex.match &lt;- ifelse(qc_info$sexp == qc_info$sex, TRUE, FALSE)
qc_info$sex.match &lt;- ifelse(samples$Inferred.Sex != samples$Sex, TRUE, FALSE)


#genotype errors
qc_info &lt;- merge(qc_info,
                 data.frame(id = names(errors_ind),
                            genotype_erros = errors_ind,stringsAsFactors = F),by = &quot;id&quot;)

##duplicated id to be remove
qc_info$remove.id.duplicated &lt;- ifelse(qc_info$id %in% summary.cg$remove.id, TRUE,FALSE)
#
bad.sample &lt;- qc_info[qc_info$generation ==1 | qc_info$Number_crossovers &lt;= 200 | qc_info$Number_crossovers &gt;=1000 | qc_info$percent_missing &gt;= 10 | qc_info$genotype_erros &gt;= 1 | qc_info$remove.id.duplicated == TRUE,]

save(qc_info, bad.sample, file = &quot;data/qc_info_bad_sample_b6.v265d.RData&quot;)

gm_samqc &lt;- gm[paste0(&quot;-&quot;,as.character(bad.sample$id)),]

#fix sex mismatch
#gm_samqc$covar[gm_samqc$covar$id %in% names(yint_ave[!names(yint_ave) %in% bad.sample$id][yint_ave[!names(yint_ave) %in% bad.sample$id] &lt;= 0.1]),&quot;sexp&quot;] &lt;- &quot;F&quot;

gm_samqc</code></pre>
<pre><code>Object of class cross2 (crosstype &quot;dof1&quot;)

Total individuals               176
No. genotyped individuals       176
No. phenotyped individuals      176
No. with both geno &amp; pheno      176

No. phenotypes                    1
No. covariates                    3
No. phenotype covariates          0

No. chromosomes                  20
Total markers                137302

No. markers by chr:
    1     2     3     4     5     6     7     8     9    10    11    12    13 
10423 10441  8206  7955  8030  8130  7760  6717  6984  6631  7433  6444  6327 
   14    15    16    17    18    19     X 
 6230  5534  5179  5323  4787  3676  5092 </code></pre>
<pre class="r"><code>save(gm_samqc, file = &quot;data/gm_samqc_b6.v265d.RData&quot;)

# update other stuff
e &lt;- e[ind_ids(gm_samqc),]
g &lt;- g[ind_ids(gm_samqc),]
#snpg &lt;- snpg[ind_ids(gm_samqc),]

#save(e,g,snpg, file = &quot;data/e_g_snpg_samqc_b6.v265d.RData&quot;)
save(e,g, file = &quot;data/e_g_snpg_samqc_b6.v265d.RData&quot;)</code></pre>
<p>Here is the list of samples that were removed:</p>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Mouse.ID
</th>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_451140_B12
</td>
<td style="text-align:center;">
451140
</td>
</tr>
<tr>
<td style="text-align:center;">
Columbia_University_John_MURGIGV01_20200529_451290_F12
</td>
<td style="text-align:center;">
451290
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_448919_A8
</td>
<td style="text-align:center;">
448919
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449250_B4
</td>
<td style="text-align:center;">
449250
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449260_G7
</td>
<td style="text-align:center;">
449260
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_449271_G4
</td>
<td style="text-align:center;">
449271
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450412_A8
</td>
<td style="text-align:center;">
450412
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450416_D10
</td>
<td style="text-align:center;">
450416
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_450428_E12
</td>
<td style="text-align:center;">
450428
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451140_E11
</td>
<td style="text-align:center;">
451140
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451290_D1
</td>
<td style="text-align:center;">
451290
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451551_E12
</td>
<td style="text-align:center;">
451551
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451556_H7
</td>
<td style="text-align:center;">
451556
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_451557_B9
</td>
<td style="text-align:center;">
451557
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_459934_A10
</td>
<td style="text-align:center;">
459934
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_459969_A4
</td>
<td style="text-align:center;">
459969
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191031_461865_F7
</td>
<td style="text-align:center;">
461865
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_448181_B3
</td>
<td style="text-align:center;">
448181
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449183_C10
</td>
<td style="text-align:center;">
449183
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:center;">
449281
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_449290_G1
</td>
<td style="text-align:center;">
449290
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_450411_H12
</td>
<td style="text-align:center;">
450411
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_451305_D11
</td>
<td style="text-align:center;">
451305
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_455737_D10
</td>
<td style="text-align:center;">
455737
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191107_459933_B1
</td>
<td style="text-align:center;">
459933
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_John_MURGIGV01_20191114_449262_E7
</td>
<td style="text-align:center;">
449262
</td>
</tr>
</tbody>
</table>
<p>Below is a table summarising the problematic samples found throughout QC. These include the following:</p>
<ul>
<li>no_pheno == sample does not have a phenotype</li>
<li>high_miss == sample has higher than 10% missing genotypes</li>
<li>diff_sex == sample has sex mismatch</li>
<li>high_xo == sample had crossover events greater than 1000 and less than 200</li>
<li>high_geno_errors == Sample has geno errors above 1%</li>
<li>DO_generation == DO generation is 1</li>
<li>dublicate_ID == sample was duplicated</li>
</ul>
<p><strong>NB:</strong> For duplcate pairs, the one that was chosen to be removed was the one that had a higher missing rate</p>
<table class="table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Mouse_ID
</th>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
Unique_Sample_ID
</th>
<th style="text-align:center;">
no_pheno
</th>
<th style="text-align:center;">
high_miss
</th>
<th style="text-align:center;">
diff_sex
</th>
<th style="text-align:center;">
high_xo
</th>
<th style="text-align:center;">
DO.Generation
</th>
<th style="text-align:center;">
high_geno.errors
</th>
<th style="text-align:center;">
duplicate_ID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;width: 3cm; ">
449260
</td>
<td style="text-align:center;width: 3cm; ">
449260
</td>
<td style="text-align:center;width: 3cm; ">
Columbia_University_John_MURGIGV01_20200529_449260_D11
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
450416
</td>
<td style="text-align:center;width: 3cm; ">
450416
</td>
<td style="text-align:center;width: 3cm; ">
Columbia_University_John_MURGIGV01_20200529_450416_B8
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451140
</td>
<td style="text-align:center;width: 3cm; ">
451140
</td>
<td style="text-align:center;width: 3cm; ">
Columbia_University_John_MURGIGV01_20200529_451140_B12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451290
</td>
<td style="text-align:center;width: 3cm; ">
451290
</td>
<td style="text-align:center;width: 3cm; ">
Columbia_University_John_MURGIGV01_20200529_451290_F12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
461865
</td>
<td style="text-align:center;width: 3cm; ">
461865
</td>
<td style="text-align:center;width: 3cm; ">
Columbia_University_John_MURGIGV01_20200529_461865_A5
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449298
</td>
<td style="text-align:center;width: 3cm; ">
449298
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20190728_449298_A10
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
448182
</td>
<td style="text-align:center;width: 3cm; ">
448182
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_448182_B5
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
448919
</td>
<td style="text-align:center;width: 3cm; ">
448919
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_448919_A8
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449187
</td>
<td style="text-align:center;width: 3cm; ">
449187
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_449187_E4
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449250
</td>
<td style="text-align:center;width: 3cm; ">
449250
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_449250_B4
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449260
</td>
<td style="text-align:center;width: 3cm; ">
449260
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_449260_G7
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449271
</td>
<td style="text-align:center;width: 3cm; ">
449271
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_449271_G4
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
450412
</td>
<td style="text-align:center;width: 3cm; ">
450412
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_450412_A8
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
450416
</td>
<td style="text-align:center;width: 3cm; ">
450416
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_450416_D10
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
450428
</td>
<td style="text-align:center;width: 3cm; ">
450428
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_450428_E12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451140
</td>
<td style="text-align:center;width: 3cm; ">
451140
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_451140_E11
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451290
</td>
<td style="text-align:center;width: 3cm; ">
451290
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_451290_D1
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451551
</td>
<td style="text-align:center;width: 3cm; ">
451551
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_451551_E12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451556
</td>
<td style="text-align:center;width: 3cm; ">
451556
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_451556_H7
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451557
</td>
<td style="text-align:center;width: 3cm; ">
451557
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_451557_B9
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
459923
</td>
<td style="text-align:center;width: 3cm; ">
459923
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_459923_E9
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
459934
</td>
<td style="text-align:center;width: 3cm; ">
459934
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_459934_A10
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
459969
</td>
<td style="text-align:center;width: 3cm; ">
459969
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_459969_A4
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
461042
</td>
<td style="text-align:center;width: 3cm; ">
461042
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_461042_G5
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
461098
</td>
<td style="text-align:center;width: 3cm; ">
461098
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_461098_C2
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
461809
</td>
<td style="text-align:center;width: 3cm; ">
461809
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_461809_A12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
461865
</td>
<td style="text-align:center;width: 3cm; ">
461865
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191031_461865_F7
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
448181
</td>
<td style="text-align:center;width: 3cm; ">
448181
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_448181_B3
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449183
</td>
<td style="text-align:center;width: 3cm; ">
449183
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_449183_C10
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449281
</td>
<td style="text-align:center;width: 3cm; ">
449281
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_449281_C12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449290
</td>
<td style="text-align:center;width: 3cm; ">
449290
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_449290_G1
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
450411
</td>
<td style="text-align:center;width: 3cm; ">
450411
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_450411_H12
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451289
</td>
<td style="text-align:center;width: 3cm; ">
451289
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_451289_C8
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
451305
</td>
<td style="text-align:center;width: 3cm; ">
451305
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_451305_D11
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
455737
</td>
<td style="text-align:center;width: 3cm; ">
455737
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_455737_D10
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
459933
</td>
<td style="text-align:center;width: 3cm; ">
459933
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191107_459933_B1
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 3cm; ">
449262
</td>
<td style="text-align:center;width: 3cm; ">
449262
</td>
<td style="text-align:center;width: 3cm; ">
Jackson_Lab_John_MURGIGV01_20191114_449262_E7
</td>
<td style="text-align:center;width: 3cm; ">
</td>
<td style="text-align:center;width: 3cm; ">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
</tbody>
</table>
<p>After full qc (sample &amp; <a href="2.3_snp_qc.html">SNP</a>), we ran all checks <a href="2.4_afterqc.html">again</a> to see if data improved for <code>R/qtl2</code> analysis.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      cluster_2.1.0     dplyr_0.8.5       optparse_1.6.6   
 [5] rhdf5_2.28.1      tidyr_1.0.2       data.table_1.12.8 fst_0.9.2        
 [9] knitr_1.28        kableExtra_1.1.0  mclust_5.4.6      ggrepel_0.8.2    
[13] ggplot2_3.3.2     qtlcharts_0.11-6  qtl2_0.22         broman_0.70-4    
[17] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      assertthat_0.2.1  rprojroot_1.3-2   digest_0.6.25    
 [5] R6_2.4.1          cellranger_1.1.0  backports_1.1.7   RSQLite_2.2.0    
 [9] evaluate_0.14     httr_1.4.1        pillar_1.4.4      rlang_0.4.6      
[13] rstudioapi_0.11   whisker_0.4       blob_1.2.1        rmarkdown_2.1    
[17] labeling_0.3      qtl_1.46-2        webshot_0.5.2     readr_1.3.1      
[21] stringr_1.4.0     htmlwidgets_1.5.1 bit_1.1-15.2      munsell_0.5.0    
[25] compiler_3.6.2    httpuv_1.5.2      xfun_0.13         pkgconfig_2.0.3  
[29] htmltools_0.4.0   tidyselect_1.0.0  tibble_3.0.1      viridisLite_0.3.0
[33] crayon_1.3.4      withr_2.2.0       later_1.0.0       grid_3.6.2       
[37] jsonlite_1.6.1    gtable_0.3.0      lifecycle_0.2.0   DBI_1.1.0        
[41] git2r_0.26.1      magrittr_1.5      scales_1.1.1      stringi_1.4.6    
[45] farver_2.0.3      fs_1.4.1          promises_1.1.0    getopt_1.20.3    
[49] xml2_1.3.1        ellipsis_0.3.1    vctrs_0.3.1       Rhdf5lib_1.6.3   
[53] tools_3.6.2       bit64_0.9-7       glue_1.4.1        purrr_0.3.4      
[57] hms_0.5.3         parallel_3.6.2    yaml_2.2.1        colorspace_1.4-1 
[61] rvest_0.3.5       memoise_1.1.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
