---
title: "Effect Of Data Cleaning [b6.v265d]"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

setwd("/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/workflowr/John-Toleman_workflow")
filepathp = '/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/Data/meta.data/'
filepaths = '/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/haplo.reconst/2020-11-11/b6.v265d/all_samples_sorted/'

#libraries
library(broman)
library(qtl2)
library(qtlcharts)
library(ggplot2)
library(ggrepel)
#library(DOQTL)
library(mclust)
#source("code/reconst_utils.R")
library("kableExtra")
library("knitr")
library("fst")
library(rhdf5)
library(ggplot2)
library(readxl) 
library(tidyr)
```

## Loading Porject
```{r loading project}
load("data/e_g_snpg_samqc_b6.v265d.RData")
gm <- get(load("data/gm_allqc_b6.v265d.RData"))

gm

````

## Samples
### Missing Data
```{r Missing data per sample, fig.height = 6, fig.width = 9.5, fig.align = "center"}

percent_missing <- n_missing(gm, "ind", "prop")*100

labels <- paste0(names(percent_missing), " (", round(percent_missing,2), "%)")
iplot(seq_along(percent_missing), percent_missing, indID=labels,
      chartOpts=list(xlab="Mouse", ylab="Percent missing genotype data",
                     ylim=c(0, 60)))

#save into pdf
pdf(file = "output/Percent_missing_genotype_data_qc_b6.v265d.pdf", width = 20, height = 20)
labels <- as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), "V01_"))[,2])
labels[percent_missing < 10] = ""
# Change point shapes and colors
p <- ggplot(data = data.frame(Mouse=seq_along(percent_missing),  
                         Percent_missing_genotype_data = percent_missing,
                         batch = factor(as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), "_"))[,5]))
                         ), 
        aes(x=Mouse, y=Percent_missing_genotype_data, color = batch)) +
        geom_point() +
        geom_hline(yintercept=10, linetype="solid", color = "red") +
        geom_text_repel(aes(label=labels), vjust = 0, nudge_y = 0.01, show.legend = FALSE, size=3) +
        theme(text = element_text(size = 10))
p

dev.off()

p

```


```{r high missing table, echo=TRUE}

save(percent_missing,
     file = "data/percent_missing_id_qc_b6.v265d.RData")

gm.covar = data.frame(id=rownames(gm$covar),gm$covar)
qc_info_cr <- merge(gm.covar,
                  data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = "id")
bad.sample.cr <- qc_info_cr[qc_info_cr$percent_missing >= 10,]

rownames(bad.sample.cr) <- NULL
bad.sample.cr[] <- lapply(bad.sample.cr, as.character)
bad.sample.cr$Neogen_Sample_ID <- sapply(strsplit(as.character(bad.sample.cr$id),'_',fixed=T),function(x) x[6])

bad.sample.cr[c("Neogen_Sample_ID","id","percent_missing")] %>%
  kable(escape = F,align = c("ccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

```{r removing bad samples CR, include=FALSE}

##removing bad samples
#gm <- gm[paste0("-",as.character(bad.sample.cr$id)),]#

#gm

```

### Sex
```{r Sexes, fig.height = 6, fig.width = 9.5, fig.align = "center"}

hdf5_filename <- dir(path = filepaths, pattern = "^hdf5_*", full.names = TRUE)
snps_file <- "/Users/corneb/Documents/MyJax/CS/Projects/QTL2_support.files/gm_uwisc_v1.csv"
snps <- read.csv(snps_file)

snps <- snps[snps$unique == TRUE, ]
#snps <- snps[snps$chr %in% c(1:19, "X"), ]
snps$chr <- sub("^chr", "", snps$chr)  ###remove prefix "chr"
colnames(snps)[colnames(snps)=="bp_mm10"] <- "pos" 
colnames(snps)[colnames(snps)=="cM_cox"] <- "cM"
snps <- snps %>% drop_na(chr, marker) 
snps$pos <- snps$pos * 1e-6
rownames(snps) <- snps$marker
colnames(snps)[1:4] <- c("marker", "chr", "pos", "pos") 

#  g <- h5read(hdf5_filename, "G")
#  g <- do.call(cbind, g)
x <- h5read(hdf5_filename, "X") # X channel intensities
x <- do.call(cbind, x)
y <- h5read(hdf5_filename, "Y") # Y channel intensities
y <- do.call(cbind, y)
rn <- h5read(hdf5_filename, "rownames")[[1]]  # markers 
cn <- h5read(hdf5_filename, "colnames")  # samples
cn <- do.call(c, cn)
# dimnames(g) <- list(rn, cn)
dimnames(x) <- list(rn, cn)
dimnames(y) <- list(rn, cn)
#cr <- colMeans(g != "--") # Call rate for each sample avg 0.95
#  sex <- determine_sex(x = x, y = y, markers = snps)$se

markers <- snps

chrx <- markers$marker[which(markers$chr == "X")]
chry <- markers$marker[which(markers$chr == "Y")]
#x[chrx,ind_ids(gm)]

chrx_int <- colMeans(x[chrx,ind_ids(gm)] + y[chrx,ind_ids(gm)], na.rm = T)
chry_int <- colMeans(x[chry,ind_ids(gm)] + y[chry,ind_ids(gm)], na.rm = T)

sample_file <- dir(path = filepaths, pattern = "^DODB_*", full.names = TRUE)
samples <- read.csv(sample_file)
samples <- samples[samples$Unique.Sample.ID %in% ind_ids(gm),]

all.equal(as.character(ind_ids(gm)), as.character(samples$Unique.Sample.ID))

#sex order
sex <- samples$Inferred.Sex

point_colors <- as.character( brocolors("web")[c("green", "purple","red")] )
percent_missing <- n_missing(gm, summary="proportion")*100
labels <- paste0(names(chrx_int), " (", round(percent_missing), "%)")
iplot( chrx_int,  chry_int, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab="Average X chr intensity", ylab="Average Y chr intensity"))
```

For figures above and below, those labelled as female in metadata given, are coloured `green`, with those labelled as male are coloured as `purple`. The above is an interactive scatterplot of the average SNP intensity on the Y chromosome versus the average SNP intensity on the X chromosome.

The cluster in the upper-left are male mice (low X chromosome intensity and high Y chromosome intensity), and the cluster in the lower-right are female mice (high X chromosome intensity and low Y chromsome intensity).  

```{r Sexes p2, fig.height = 6, fig.width = 9.5, fig.align = "center"}

phetX <- rowSums(gm$geno$X == 2)/rowSums(gm$geno$X != 0)
phetX <- phetX[names(phetX) %in% names(chrx_int)]
iplot(chrx_int, phetX, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab="Average X chr intensity", ylab="Proportion het on X chr"))
```

In the above scatterplot, we show the proportion of hets vs the average intensity for the X chromosome SNPs. In calculating the proportion of heterozygous genotypes for the individuals, we look at X chromosome genotypes equal to 2 which corresponds to the heterozygote) relative to not being 0 (which is used to encode missing genotypes). The genotypes are arranged with rows being individuals and columns being markers. 

The following are the mice that have had sex incorrectly assigned:

```{r sex diff table, echo=FALSE}

bad.sample.sex <- subset(samples, (samples$Inferred.Sex != samples$Inferred.Sex) | is.na(samples$Sex))
#bad.sample.sex$Sex[is.na(bad.sample.sex$Sex)] <- '--'
bad.sample.sex$Neogen_Sample_ID <- sapply(strsplit(as.character(bad.sample.sex$Unique.Sample.ID),'_',fixed=T),function(x) x[6])
rownames(bad.sample.sex) <- NULL

bad.sample.sex[c("Neogen_Sample_ID","Unique.Sample.ID","Sex","Inferred.Sex")] %>%
  kable(escape = F,align = c("ccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

### Sample Duplicates
```{r Sample duplicates}

cg <- compare_geno(gm, cores=10)
summary.cg <- summary(cg)
```

```{r Sample duplicates table, echo=TRUE}
rownames(summary.cg) <- NULL
summary.cg$iid1 <- sapply(strsplit(as.character(summary.cg$ind1),'_',fixed=T),function(x) x[6]) 
summary.cg$iid2 <- sapply(strsplit(as.character(summary.cg$ind2),'_',fixed=T),function(x) x[6]) 

summary.cg[c(1,9,2,10,3:6)] %>%
  kable(escape = F,align = c("llcccccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F) 

```

Here is a histogram of the proportion of matching genotypes. The tick marks below the histogram indicate individual pairs.

```{r Sample duplicates figures}

save(summary.cg,
     file = "data/summary.cg_qc_b6.v265d.RData")

pdf(file = "output/Proportion_matching_genotypes_before_removal_of_bad_samples_qc_b6.v265d.pdf", width = 20, height = 20) 
par(mar=c(5.1,0.6,0.6, 0.6))
hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
     main="", yaxt="n", ylab="", xlab="Proportion matching genotypes")
rug(cg[upper.tri(cg)])
dev.off()

par(mar=c(5.1,0.6,0.6, 0.6))
hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
     main="", yaxt="n", ylab="", xlab="Proportion matching genotypes")
rug(cg[upper.tri(cg)])

```

```{r Sample duplicates 2 figures, include=FALSE}

#cgsub <- cg[percent_missing < 50, percent_missing < 50]
#par(mar=c(5.1,0.6,0.6, 0.6))
#hist(cgsub[upper.tri(cgsub)], breaks=seq(0, 1, length=201),
#     main="", yaxt="n", ylab="", xlab="Proportion matching genotypes")
#rug(cgsub[upper.tri(cgsub)])

```

```{r top 20 high missing rate, include = FALSE}
##show top 20 samples with missing genotypes
#percent_missing <- n_missing(gm, "ind", "prop")*100
#round(sort(percent_missing, decreasing=TRUE)[1:20], 1)  
```

```{r removing duplicate samples, include=FALSE}
#removing duplcate pairs

pheno_file_nick <- dir(path = filepathp, pattern = "*_pheno.csv+$", full.names = TRUE)
pheno.pi <- read.csv(pheno_file_nick)

master <- as.data.frame(read_excel("/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/Data/meta.data/inventory_file.xlsx", sheet = "inventory_file"))

#sample_file <- dir(path = filepaths, pattern = "^DODB_*", full.names = TRUE)
#samples <- read.csv(sample_file)

project <- read.csv("/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/haplo.reconst/2020-11-09/all.samples/all.samples/project_b6.v265d.csv", as.is=T)
project$dir2 <- sapply(strsplit(project$dir,'/',fixed=T),function(x) x[7])

inventory <- master[master$"Sample ID...4" %in% samples$Original.Mouse.ID,]
invent.dir2 <- merge(inventory,project, by.x=c("Directory"), by.y=c("dir2"), all=T)
invent.dodb <- merge(invent.dir2,samples, by.x=c("Directory","Sample ID...4"), by.y=c("Directory","Original.Mouse.ID"),all=T)

gm.pheno1 <- merge(invent.dodb,pheno.pi, by.x=c("Sample ID...5"), by.y=c("Mouse.ID"), all.x=T)
gm.pheno <- gm.pheno1[c("Correct.Mouse.ID","Unique.Sample.ID","Unique Sample ID","Sample ID...4","Sample ID...5","Directory",
                     "Sex.x","Sex.y","Sex","Inferred.Sex","Birth Date","DOB","DO Generation","DO.Mother.Generation","DO.Generation","Strain.x","Strain.y",
                     "Lmx1b allele","Lmx1b.allele","Lmx1b Genotype","Lmx1b.Genotype","Parental.Strain.GigaMUGA.ID","Parental Strain GigaMUGA ID",
                     "Run.Date","Geneseek.Sample.ID","Include","cr","Max_IOP")]

summary.cg$Name.ind1 <- as.character(do.call(rbind.data.frame, strsplit(as.character(summary.cg$ind1), "_"))[,6])
summary.cg$Name.ind2 <- as.character(do.call(rbind.data.frame, strsplit(as.character(summary.cg$ind2), "_"))[,6])
summary.cg$miss.ind1 <- percent_missing[match(summary.cg$ind1, names(percent_missing))]
summary.cg$miss.ind2 <- percent_missing[match(summary.cg$ind2, names(percent_missing))]
summary.cg$phen.ind1 <- gm.pheno$Max_IOP[match(summary.cg$ind1, gm.pheno$Unique.Sample.ID)]
summary.cg$phen.ind2 <- gm.pheno$Max_IOP[match(summary.cg$ind2, gm.pheno$Unique.Sample.ID)]
summary.cg$remove.id <- ifelse((summary.cg$miss.ind1 > summary.cg$miss.ind2), summary.cg$ind1, summary.cg$ind2)
#summary.cg$remove.id.p1 <- ifelse(is.na(summary.cg$phen.ind1) & (is.na(summary.cg$phen.ind1) | !is.na(summary.cg$phen.ind1)), summary.cg$ind1, '')
#summary.cg$remove.id.p2 <- ifelse(is.na(summary.cg$phen.ind2) & (is.na(summary.cg$phen.ind1) | !is.na(summary.cg$phen.ind1)), summary.cg$ind2, '')
#remove.IDs <- rbind(summary.cg$remove.id.m[-1],summary.cg$remove.id.p1[-1],summary.cg$remove.id.p2[-1])
summary.cg$remove.id  

#qc_info$remove.id.duplicated <- ifelse(qc_info$id %in% summary.cg$remove.id, TRUE,FALSE)

#bad.sample.cr.dup <- NULL

#bad.sample.cr.dup$id <- c("Univ_of_Georgia_Pazdro_MURMUGV01_20200130_9_A6",#
#	                     "Univ_of_Penn_Thiass_MURGIGV01_20200226_90_C12", 
#	                     "Univ_of_Georgia_Pazdro_MURMUGV01_20200130_109_G2",
#	                     "Univ_of_Georgia_Pazdro_MURMUGV01_20200130_111_B3",
#	                     "Univ_of_Georgia_Pazdro_MURMUGV01_20200130_110_H5",
#	                     "Univ_of_Georgia_Pazdro_MURMUGV01_20200130_112_F5")

#gm <- gm[paste0("-",as.character(bad.sample.cr.dup$id)),]

#gm

#percent_missing <- n_missing(gm, "ind", "prop")*100
#round(sort(percent_missing, decreasing=TRUE)[1:19], 1)

```

### Array Intensities
```{r Array intensities, fig.height = 6, fig.width = 9.5, fig.align = "center"}

#load the intensities.fst_qc.RData
#load("data/intensities.fst.RData")
#X and Y channel
xn <- x[,ind_ids(gm)]
xn <- xn[snps$marker,]
xnm <- rownames(xn)

yn <- y[,ind_ids(gm)]
yn <- yn[snps$marker,]

# bring together in one matrix
result <- cbind(snp=rep(snps$marker, 2),
                channel=rep(c("x", "y"), each=length(snps$marker)),
                as.data.frame(rbind(xn, yn)))
rownames(result) <- 1:nrow(result)

# bring SNP rows together
result <- result[as.numeric(t(cbind(seq_along(snps$marker), seq_along(snps$marker)+length(snps$marker)))),]
rownames(result) <- 1:nrow(result)

#load the intensities.fst.RData
#load("data/heh/intensities.fst.RData")
#X and Y channel
X <- result[result$channel == "x",]
rownames(X) <- X$snp
X <- X[,c(-1,-2)]

Y <- result[result$channel == "y",]
rownames(Y) <- Y$snp
Y <- Y[,c(-1,-2)]

int <- result

#int <- result

#rm(result)
int <- int[seq(1, nrow(int), by=2),-(1:2)] + int[-seq(1, nrow(int), by=2),-(1:2)]
int <- int[,intersect(ind_ids(gm), colnames(int))]
n <- names(sort(percent_missing[intersect(ind_ids(gm), colnames(int))], decreasing=TRUE))
iboxplot(log10(t(int[,n])+1), orderByMedian=FALSE, chartOpts=list(ylab="log10(SNP intensity + 1)"))
```
In the above plot, distributions of array intensities (after a log10(x+1) transformation) are displayed. 

The arrays are sorted by the proportion of missing genotype data for the sample, and the curves connect various quantiles of the intensities.

```{r Array intensities percentilw figure, fig.height = 6, fig.width = 9.5, fig.align = "center"}

qu <- apply(int, 2, quantile, c(0.01, 0.99), na.rm=TRUE)
group <- (percent_missing >= 19.97) + (percent_missing > 5) + (percent_missing > 2) + 1
labels <- paste0(colnames(qu), " (", round(percent_missing), "%)")
iplot(qu[1,], qu[2,], indID=labels, group=group,
      chartOpts=list(xlab="1 %ile of array intensities",
                     ylab="99 %ile of array intensities",
                     pointcolor=c("#ccc", "slateblue", "Orchid", "#ff851b")))
```
For this particular set of arrays, a plot of the 1 %ile vs the 99 %ile is quite revealing. In the following, the orange points are those with > 20% missing genotypes, the pink points are the samples with 5-20% missing genotypes, and the blue points are the samples with 2-5% missing genotypes.

### Genotype Frequencies

The following triangle plots show the genotype frequency distributions for the mice, among the four groups of markers with common minor allele frequency (MAF) in the founder strains. These plots make use of the fact that for a point within an equilateral triangle, the sum of the distances to the three sides is a constant.

```{r Genotype frequencies}
g <- do.call("cbind", gm$geno[1:19])
fg <- do.call("cbind", gm$founder_geno[1:19])
g <- g[,colSums(fg==0)==0]
fg <- fg[,colSums(fg==0)==0]
fgn <- colSums(fg==3)

gf_ind <- vector("list", 4)
for(i in 1:4) {
  gf_ind[[i]] <- t(apply(g[,fgn==i], 1, function(a) table(factor(a, 1:3))/sum(a != 0)))
}

par(mfrow=c(4,1), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c("AA", "AB", "BB"), main=paste0("MAF = ", i, "/8"))
  tripoints(gf_ind[[i]], pch=21, bg="lightblue")
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg="violetred")
  
  if(i>=3) { # label mouse with lowest het
    wh <- which(gf_ind[[i]][,2] == min(gf_ind[[i]][,2]))
    tritext(gf_ind[[i]][wh,,drop=FALSE] + c(0.02, -0.02, 0),
            names(wh), adj=c(0, 1))
  }
  
  # label other mice
  if(i==1) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.3]
  }
  else if(i==2) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.48]
  }
  else if(i==3) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.51]
  }
  else if(i==4) {
    lab <- rownames(gf_ind[[i]])[gf_ind[[i]][,2]>0.6]
  }
  
  for(ind in lab) {
    if(grepl("^F", ind) && i != 3) {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(-0.01, 0, +0.01), ind, adj=c(1,0.5))
    } else {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(0.01, 0, -0.01), ind, adj=c(0,0.5))
    }
  }
}

````

The majority of individuals are tightly clustered around the expected distribution (in pink). 

```{r removing low af, include=FALSE}

#bad.sample.cr.dup.gf <- NULL

#bad.sample.cr.dup.gf$id <- c("Univ_of_Georgia_Pazdro_MURMUGV01_20200130_108_D12")

#gm <- gm[paste0("-",as.character(bad.sample.cr.dup.gf$id)),]

#gm

```

### Crossover Counts
```{r Crossover counts, fig.height = 6, fig.width = 9.5, fig.align = "center"}
#load pre-caluated results
pr<-readRDS("/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/haplo.reconst/2020-11-11/b6.v265d/all_samples_sorted/Jax.Columbia_John__GigaMUGA_genoprobs_36state_sorted.rds")
#load("data/m.RData")
#load("data/nxo.RData")

m <- maxmarg(pr)
nxo <- count_xo(m)
#save(pr, file = "data/pr.RData")
#str(pr)
save(m, file = "data/m_qc_b6.v265d.RData")
save(nxo, file = "data/nxo_qc_b6.v265d.RData")

e <- calc_errorlod(gm, pr)
#str(e)
e <- do.call("cbind", e)
save(e, file = "data/e_qc_b6.v265d.RData")

#snpg <- predict_snpgeno(gm, m)
#snpg <- do.call("cbind", snpg)
#save(snpg, file = "data/snpg_b6.v265d.RData")

#crossover
totxo <- rowSums(nxo)[rownames(gm$covar)]
iplot(seq_along(totxo),
      totxo,
      group=gm$covar$generation,
      chartOpts=list(xlab="Mouse", ylab="Number of crossovers", 
                     margin=list(left=80,top=40,right=40,bottom=40,inner=5),
                     axispos=list(xtitle=25,ytitle=50,xlabel=5,ylabel=5)))

#save crossover into pdf
pdf(file = "output/number_crossover_qc_b6.v265d.pdf")
cross_over <- data.frame(Mouse = seq_along(totxo), Number_crossovers = totxo, generation = gm$covar$generation)
names(totxo) <- as.character(do.call(rbind.data.frame, strsplit(names(totxo), "V01_"))[,2])
names(totxo)[totxo >= 200 & totxo <= 800] = ""
# Change point shapes and colors
p <-ggplot(cross_over, aes(x=Mouse, y=Number_crossovers, fill = generation, color=generation)) +
  geom_point() +
  geom_text_repel(aes(label=names(totxo),hjust=0,vjust=0), show.legend = FALSE)
p
dev.off()

p

```
Counts of inferred crossovers can be a useful diagnostic for identifying problem samples, which may show an excessive number of apparent crossovers. Above is a plot of the number of crossovers vs the mouse index, colored by generation.

```{r  corssover counts, echo = FALSE}
#Here are the crossover counts for those  mice:
tmp <- cbind(percent_missing=round(percent_missing[rownames(gm$covar)],2), total_xo=totxo)[percent_missing[rownames(gm$covar)] >= 5,]
if(length(tmp) >2 ) {
  tmp[order(tmp[,1]),]
} else {
  tmp
}

##number of crossovers

totxo <- rowSums(nxo)[rownames(gm$covar)]
totxot <- totxo[totxo <= 200 | totxo >= 1000]

totxot <- as.data.frame(totxot)
#totxot <- as.data.frame(t(totxot))

totxot$Neogen_Sample_ID <- sapply(strsplit(rownames(totxot),'_',fixed=T),function(x) x[6]) 
totxot$number.xo <- totxot$totxot
totxot$id <- rownames(totxot)

rownames(totxot) <- NULL

totxot[c(2,4,3)] %>%
  kable(escape = F,align = c("cc"),linesep ="\\hline") %>%
  kable_styling(full_width = F)  %>%
  kable_styling("striped", full_width = F) 

```

### Genotyping Error LOD Scores
```{r  Genotyping error LOD scores, fig.height = 6, fig.width = 9.5, fig.align = "center"}
load("data/e_qc_b6.v265d.RData")
errors_ind <- rowSums(e>2)[rownames(gm$covar)]/n_typed(gm)*100
lab <- paste0(names(errors_ind), " (", myround(percent_missing[rownames(gm$covar)],1), "%)")
iplot(seq_along(errors_ind), errors_ind, indID=lab,
      chartOpts=list(xlab="Mouse", ylab="Percent genotyping errors", ylim=c(0, 8),
                     axispos=list(xtitle=25, ytitle=50, xlabel=5, ylabel=5)))
save(errors_ind, file = "data/errors_ind_qc_b6.v265d.RData")

```


### Apparent Genotyping Errors
```{r apparent genotyping errors, fig.height = 6, fig.width = 9.5, fig.align = "center", include=FALSE}

#load("data/snpg_b6.v265d.RData")

snpgn <- do.call("cbind", gm$geno)

gobs <- do.call("cbind", gm$geno)
gobs[gobs==0] <- NA

par(pty="s")
err_direct <- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 <- rowSums(e > 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab="Percent errors (error LOD > 0)",
         ylab="Percent errors (obs vs predicted)",
         xlim=c(0, 2), ylim=c(0, 2))
abline(0,1,lty=2, col="gray60")

pdf(file = "output/Percent_genotype_errors_obs_vs_predicted_b6.v265d.pdf",width = 20, height = 20) 
par(pty="s")
err_direct <- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 <- rowSums(e > 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab="Percent errors (error LOD > 0)",
         ylab="Percent errors (obs vs predicted)",
         xlim=c(0, 2), ylim=c(0, 2))
abline(0,1,lty=2, col="gray60")
dev.off()

par(pty="s")
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab="Percent errors (error LOD > 0)",
         ylab="Percent errors (obs vs predicted)",
         xlim=c(0, 0.4), ylim=c(0, 0.4))
abline(0,1,lty=2, col="gray60")

```

## Markers

### Marker Missing Data 
```{r Missing data in Markers and Genotype frequencies Markers, fig.height = 6, fig.width = 9.5, fig.align = "center"}

pmis_mar <- n_missing(gm, "marker", "proportion")*100
save(pmis_mar, file = "data/percent_missing_marker_qc_b6.v265d.RData")

par(mar=c(5.1,0.6,0.6, 0.6))
hist(pmis_mar, breaks=seq(0, 100, length=201),
     main="", yaxt="n", ylab="", xlab="Percent missing genotypes")
rug(pmis_mar)

pdf(file = "output/Percent_missing_genotype_data_per_marker_qc_b6.v265d.pdf")
par(mar=c(5.1,0.6,0.6, 0.6))
hist(pmis_mar, breaks=seq(0, 100, length=201),
     main="", yaxt="n", ylab="", xlab="Percent missing genotypes")
rug(pmis_mar)
dev.off()
```

```{r missing table, echo=FALSE}

pmis <- NULL
pmis$pmis_mar_5 <- sum(pmis_mar >= 5)
pmis$pmis_mar_10 <- sum(pmis_mar >= 10)
pmis$pmis_mar_15 <- sum(pmis_mar >= 15)
pmis$pmis_mar_25 <- sum(pmis_mar >= 25)
pmis$pmis_mar_50 <- sum(pmis_mar >= 50)
pmis$pmis_mar_75 <- sum(pmis_mar >= 75)
pmis$total_snps <- nrow(as.data.frame(pmis_mar))

pmis <- t(as.data.frame(pmis))
pmis <- as.data.frame(pmis)
pmis$count <- pmis$V1

pmis[c(2)] %>%
  kable(escape = F,align = c("ccccccccc"),linesep ="\\hline") %>%
  kable_styling(full_width = F) %>%
  kable_styling("striped", full_width = F)  %>%
  row_spec(7 ,bold=T,color= "white",background = "black")

```

For the markers with lots of missing genotypes, itâ€™s not necessarily the case that the remaining genotypes are full of errors, but in studying the allele intensities at these SNPs, it does appear that for the bulk of such markers, the genotypes are not being called appropriately.

### Marker Genotype Frequencies
```{r Genotype frequencies Markers, fig.height = 6, fig.width = 9.5, fig.align = "center"}

g <- do.call("cbind", gm$geno[1:19])
fg <- do.call("cbind", gm$founder_geno[1:19])
g <- g[,colSums(fg==0)==0]
fg <- fg[,colSums(fg==0)==0]
fgn <- colSums(fg==3)

gf_mar <- t(apply(g, 2, function(a) table(factor(a, 1:3))/sum(a != 0)))
gn_mar <- t(apply(g, 2, function(a) table(factor(a, 1:3))))

pdf(file = "output/genotype_frequency_marker_qc_b6.v265d.pdf")
par(mfrow=c(2,2), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c("AA", "AB", "BB"), main=paste0("MAF = ", i, "/8"))
  z <- gf_mar[fgn==i,]
  z <- z[rowSums(is.na(z)) < 3,]
  tripoints(z, pch=21, bg="gray80", cex=0.6)
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg="violetred")
}
dev.off()

par(mfrow=c(2,2), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c("AA", "AB", "BB"), main=paste0("MAF = ", i, "/8"))
  z <- gf_mar[fgn==i,]
  z <- z[rowSums(is.na(z)) < 3,]
  tripoints(z, pch=21, bg="gray80", cex=0.6)
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg="violetred")
}

save(gf_mar, file = "data/genotype_freq_marker_qc_b6.v265d.RData")

```

The bulk of the markers seem well behaved, but there are a number of markers with unusual genotype frequencies. There are markers that show no homozygotes for the major allele. (These sit on the left edge.) There are markers that are monomorphic (100% AA genotypes; lower-right vertex). And there some markers with some of each homozygote but no heterozygotes (bottom edge). Further, there are a bunch of markers where the minor allele appears to be private to one founder strain (upper-left panel) but show a high frequency of minor alleles in the DO offspring. 


### Marker Genotype Errors
```{r Genotype errors Markers, fig.height = 6, fig.width = 9.5, fig.align = "center"}
#load("data/e_g_snpg_samqc.RData")

markers <- colnames(do.call("cbind", gm$geno))
en <- e[ , markers]

errors_mar <- colSums(en>2)/n_typed(gm, "marker")*100

grayplot(pmis_mar, errors_mar,
         xlab="Proportion missing", ylab="Proportion genotyping errors")

pdf(file = "output/genotype_error_marker_qc_b6.v265d.pdf")
grayplot(pmis_mar, errors_mar,
         xlab="Proportion missing", ylab="Proportion genotyping errors")
dev.off()

save(errors_mar, file = "data/genotype_errors_marker_qc_b6.v265d.RData")

```

Markers with higher rates of missing genotypes tend to show higher errors rates.

### Array Intensities


### Removing Markers
#### Missingness
```{r listing markers missing, echo=FALSE}

#Missingness

length(pmis_mar[pmis_mar >= 10])

#high_miss <- find_markerpos(gm, names(pmis_mar[pmis_mar >= 10]))
#high_miss$id <- rownames(high_miss)
#high_miss_df <- as.data.frame(pmis_mar[pmis_mar >= 10])
#high_miss_df$index = 1: nrow(high_miss_df)
#high_miss_df$id <- rownames(high_miss_df)

#high_miss_bad <- merge(high_miss,high_miss_df, by=c("id"),all=T)
#names(high_miss_bad)[5] <- c("high_miss")
#names(high_miss_bad)[1] <- c("marker")
#high_miss_bad <- high_miss_bad[order(high_miss_bad$index),]
```

#### Monomorphic/Low Frequency markers
```{r listing markers Monomorphic or Low Frequency markers, echo=FALSE}

#Monomorphic/Low Frequency markers

low_freq_df <- as.data.frame(gf_mar)
low_freq_df$count <- rowSums(low_freq_df <= 0.01)
low_freq_df <- low_freq_df[low_freq_df$count == 2,]
low_freq_df$id <- rownames(low_freq_df)
#low_freq_df$index = 1: nrow(low_freq_df)
low_freq <- find_markerpos(gm, rownames(low_freq_df))
low_freq$id <- rownames(low_freq)

nrow(low_freq)

#low_freq_bad <- merge(low_freq,low_freq_df, by=c("id"),all=T)
#names(low_freq_bad)[5] <- c("AA_freq")
#names(low_freq_bad)[6] <- c("AB_freq")
#names(low_freq_bad)[7] <- c("BB_freq")
#names(low_freq_bad)[1] <- c("marker")
#low_freq_bad <- low_freq_bad[order(low_freq_bad$index),]

```

#### Genotyping Error
```{r listing markers Genotyping Error, echo=FALSE}

##Genotyping Error

length(errors_mar[errors_mar > 5])

#error_markers <- find_markerpos(gm, names(errors_mar[errors_mar > 5]))
#error_markers$id <- rownames(error_markers)
#rne <- rownames(as.data.frame(errors_mar))
#error_mars_df <- as.data.frame(errors_mar[errors_mar > 5])
#error_mars_df$id = rownames(error_mars_df)
#error_mars_df$index = 1: nrow(error_mars_df)

#error_markers_bad <- merge(error_markers,error_mars_df, by=c("id"),all=T)
#names(error_markers_bad)[5] <- c("error_mars")
#names(error_markers_bad)[1] <- c("marker")
#error_markers_bad <- error_markers_bad[order(error_markers_bad$index),]

```

#### Total
```{r listing markers total, echo=FALSE}

### merge all

#bad_markers <- rbind(high_miss_bad[c("marker","chr","gmap","pmap")], low_freq_bad[c("marker","chr","gmap","pmap")], error_markers_bad[c("marker","chr","gmap","pmap")])
nrow(low_freq)

#duplicate <- bad_markers[duplicated(bad_markers),]

#bad_markers <- bad_markers[!duplicated(bad_markers),]
#nrow(bad_markers)

#save(bad_markers, file = "data/bad_markers_all_qc.RData")

#bad_markers <- find_markerpos(gm, names(errors_mar[errors_mar > 5]))
#save(bad_markers, file = "data/bad_markers_qc.RData")

```
