---
title: "John/Tolman Workflow"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
---

### Project At A Glance
* Samples: b6.v265d
	+ [Total Samples: 202](0.1_samples_batch_b6.v265d.html)
	+ [Total Samples (After Correcting For Problematic Samples): 185](0.1_samples_batch_b6.v265d_aqc.html)
* Samples: b6.q105x
	+ [Total Samples: 383](0.1_samples_batch_b6.q105x.html)
	+ [Total Samples (After Correcting For Problematic Samples): 375](0.1_samples_batch_b6.q105x_aqc.html)
* Samples: d2.q105x
	+ [Total Samples: 134](0.1_samples_batch_d2.q105x.html)
	+ [Total Samples (After Correcting For Problematic Samples): 131](0.1_samples_batch_d2.q105x_aqc.html)
* Samples: 129Sj.v265d
	+ [Total Samples: 136](0.1_samples_batch_129Sj.v265d.html)
	+ [Total Samples (After Correcting For Problematic Samples): 127](0.1_samples_batch_129Sj.v265d_aqc.html)

### Haplotype Phasing
* Before QC
	+ [b6.v265d](0.2_haplotype_comparison_bqc_b6.v265d.html)
	+ [b6.q105x](0.2_haplotype_comparison_bqc_b6.q105x.html)
	+ [d2.q105x](0.2_haplotype_comparison_bqc_d2.q105x.html)
	+ [129Sj.v265d](0.2_haplotype_comparison_bqc_129Sj.v265d.html)
* After Correcting For Problematic Samples
	+ [b6.v265d](0.2_haplotype_comparison_aqc_b6.v265d.html)
	+ [b6.q105x](0.2_haplotype_comparison_aqc_b6.q105x.html)
	+ [d2.q105x](0.2_haplotype_comparison_aqc_d2.q105x.html)
	+ [129Sj.v265d](0.2_haplotype_comparison_aqc_129Sj.v265d.html)

### 1. Phenotype QC
* Phenotypes - Max_IOP
	+ [b6.v265d](1.1_phenotypes_b6.v265d.html)
	+ [b6.q105x](1.1_phenotypes_b6.q105x.html)
	+ [d2.q105x](1.1_phenotypes_d2.q105x.html)
	+ [129Sj.v265d](1.1_phenotypes_129Sj.v265d.html)

### 2. Genotype QC
* Step 1: Sample QC 
	+ Before QC
		+ [b6.v265d](2.1_sample_bqc_b6.v265d.html)
		+ [b6.q105x](2.1_sample_bqc_b6.q105x.html)
		+ [d2.q105x](2.1_sample_bqc_d2.q105x.html)
		+ [129Sj.v265d](2.1_sample_bqc_129Sj.v265d.html)
		+ [all samples](2.1_sample_bqc_all.samples.html)
	+ After Correcting For Problematic Samples
		+ [b6.v265d](2.1_sample_bqc_b6.v265d_aqc.html)
		+ [b6.q105x](2.1_sample_bqc_b6.q105x_aqc.html)
		+ [d2.q105x](2.1_sample_bqc_d2.q105x_aqc.html)
		+ [129Sj.v265d](2.1_sample_bqc_129Sj.v265d_aqc.html)
* Step 2: SNP QC (Monomorphic, Genotype Error & Missing Data)
	+ Before QC
		+ [b6.v265d](2.2.1_snp_qc_b6.v265d.html)
		+ [b6.q105x](2.2.1_snp_qc_b6.q105x.html)
		+ [d2.q105x](2.2.1_snp_qc_d2.q105x.html)
		+ [129Sj.v265d](2.2.1_snp_qc_129Sj.v265d.html)
		+ [all samples](2.2.1_snp_qc_all.samples.html)
	+ After Correcting For Problematic Samples
		+ [b6.v265d](2.2.1_snp_qc_b6.v265d_aqc.html)
		+ [b6.q105x](2.2.1_snp_qc_b6.q105x_aqc.html)
		+ [d2.q105x](2.2.1_snp_qc_d2.q105x_aqc.html)
		+ [129Sj.v265d](2.2.1_snp_qc_129Sj.v265d_aqc.html)
		+ [all samples]()
* Step 3: Effect of Data Cleaning
	+ Before QC
		+ [b6.v265d](2.3.1_afterqc_b6.v265d.html)
		+ [b6.q105x](2.3.1_afterqc_b6.q105x.html)
		+ [d2.q105x](2.3.1_afterqc_d2.q105x.html)
		+ [129Sj.v265d](2.3.1_afterqc_129Sj.v265d.html)
		+ [all samples](2.3.1_afterqc_all.samples.html)
	+ After Correcting For Problematic Samples
		+ [b6.v265d](2.3.1_afterqc_b6.v265d_aqc.html)
		+ [b6.q105x](2.3.1_afterqc_b6.q105x_aqc.html)
		+ [d2.q105x](2.3.1_afterqc_d2.q105x_aqc.html)
		+ [129Sj.v265d](2.3.1_afterqc_129Sj.v265d_aqc.html)
		+ [all samples]()
* Step 4: Preparing Files for QTL Analysis
	+ [After Correcting For Problematic Samples](2.4.1_preparing_files_qtl2.html)


### 3. QTL Analysis
* Step 1: Recombination Block Size & Founder Proportions
* Step 2: GCTA Heritability
* Step 3: QTL Mapping (`R/qtl2`)
* Step 4: QTL Permutation (`R/qtl2`)
* Step 5: QTL Best Linear Unbiased Prediction (BLUP) (`R/qtl2`)
* Step 6: Results from `R/qtl2` analysis [Raw Data]
	+ All variants included
		+ [max IOP](3.6_plot_qtl_mapping_max_IOP_sorted.html)
		+ [max corneal haze](3.6_plot_qtl_mapping_max_corneal_haze_sorted.html)
		+ [Pupil_not_round_binary](3.6_plot_qtl_mapping_Pupil_not_round_binary_sorted.html)
